---
title: "STJ_SCTLD_Timeseries_Phyloseq_Sept2023"
author: "JeanneBloomberg"
date: "5/27/2023"
output: html_document
---

#Set up
```{r setup, include=FALSE}
setwd("~/Documents/WHOI/Projects/STJ_SCTLD_TS")
```

```{r}
# For data wrangling
library(dplyr); packageVersion("dplyr")
library(tidyr); packageVersion("tidyr")

# For microbiome/metagenome analysis
library(corncob)
library(vegan); packageVersion("vegan")
library(phyloseq); packageVersion("phyloseq")

# For visualization
library(ggplot2); packageVersion("ggplot2")
library(randomcoloR)

```

#Upload data
```{r}
load("~/Documents/WHOI/Projects/STJ_SCTLD_TS/otus_SCTSCTLDTD_J1N2V2_final.RData")

taxa <- as.matrix(read.table("taxonomy_SCTSCTLDTD_J1N2V2_final.txt", sep = "\t", row.names = 1, header = TRUE))

ASV_ID <-read.table("ASVsequences_SCTSCTLDTD_J1N2V2_final.txt", sep = "\t", row.names = 1, header = TRUE)

metadata <- read.csv("meta_table.csv", header = TRUE)
metadata <- metadata[,1:26]
row.names(metadata) <- metadata$Sample_names

```

#Removes <10,000 reads
Added this processing on Nov. 3, 2023 (orginially in DADA2_STJ_SCTLD_Timeseries_Aug2023.Rmd, but made more sense to me to put it here this time.)
This code is for a special case. In the field, one syringe of seawater was filtered through two different filters, so the the two sequences are actually representative of one sample. I will combine the two samples into one. 
```{r}
#SW 32 and SW 33 are the same water sample, split between two filters
#So I am going to combine the counts from SW-33 to SW-32
head(otus["SW-32",])
  # ASV1 ASV2 ASV3 ASV4 ASV5 ASV6 
  # 882   20   79    0   59   25 
head(otus["SW-33",])
  # ASV1 ASV2 ASV3 ASV4 ASV5 ASV6 
  # 99    0    0    0    0    0 

otus["SW-32",] <- otus["SW-32",] + otus["SW-33",]
head(otus["SW-32",])
    # ASV1 ASV2 ASV3 ASV4 ASV5 ASV6 
    # 981   20   79    0   59   25 
    # OK I think it worked, now I have to remove sample 33 from every phyloseq with Fish Bay Seawater
```


Goal: remove samples with less than 10,000 reads
```{r}

#look at the total reads for each sample
total_reads_per_sample <- data.frame(Sample = row.names(otus),
                                        Reads = rowSums(otus[,]))
keep <- rowSums(otus[,]) >= 10000

removed <- rowSums(otus[,]) < 10000

Samples_removed <- as.data.frame(removed[removed == TRUE])

otus_noLowReads <- otus[keep,]

cat(sum(removed, na.rm = TRUE), 'samples were removed because they had less than 10000 sequences.\n')
  # 25 samples were removed because they had less than 10000 sequences.
  # (but of these, 19 are negative controls)

cat(sum(keep, na.rm = TRUE), 'samples have sufficient reads and thus, remain in the dataset.\n')
  # 194 samples have sufficient reads and thus, remain in the dataset.
```
The non-control samples that are removed are: 
Coral 11 --> losing 
Coral 107 --> losing
SW 33 --> this is a half sample with SW-32, so not losing a data point
SW 34 --> this has a duplicate, so not losing a data point
SW 42 --> this has a duplicate, so not losing a data point
SW 9 --> losing


#Remove duplicates

Some of the samples have duplicates, so I want to pick which duplicate is better to process. 
I am choosing the sample with more reads, but I think it only matters when there are orders of magnitudes in difference. 
  ^^ I think this isn't what I'm supposed to do with these samples, talk to Amy (note on 8/21/24)
```{r}
TotalReads <- as.data.frame(rowSums(otus_noLowReads[,]))

#Manually checked which samples has more reads for each remaining duplicate
#Going to remove: 
#  Coral_52_1
#  SW-26
#  SW-35
#  SW-8duplicate

duplicates_toRemove <- c("Coral_52_1", "SW-26", "SW-35", "SW-8duplicate")

otus_noDuplicates <- otus_noLowReads[!(rownames(otus_noLowReads) %in% duplicates_toRemove),]

dim(otus)
  # 219 23901
dim(otus_noLowReads)
  # 194 23901
dim(otus_noDuplicates)
  # 190 23901
```


Ok now I am going to cut down the meta data so that it only includes the samples I am using going forward -- i.e. without samples with low read yields and without duplicates

```{r}
otus <- otus_noDuplicates
samples_we_use <- row.names(otus)
ASV <- as.matrix(otus)

metadata <- metadata[samples_we_use, ]

```


#Make base phyloseq object 

```{r}
# Make a phyloseq object. 
ASV = otu_table(ASV, taxa_are_rows = FALSE)
TAX = tax_table(taxa)
META = sample_data(metadata)

ps <- phyloseq(ASV, TAX, META)
  #taxa: 23901
  #samples: 190

#remove contaminants
load("~/Documents/WHOI/Projects/STJ_SCTLD_TS/contam.RData")

ps.decontam <- prune_taxa(!contam$contaminant, ps)#went from 20481 taxa to 20430 taxa
  #taxa: 23857
  #samples: 190

#remove ASVS that are mitochondria or chloroplasts
ps.decontam.nomitochloro <- ps.decontam %>%
    subset_taxa(Family !="Mitochondria" & Order !="Chloroplast")
  #taxa: 22618
  #samples: 190

#remove ASVS with all zeros 
ps.decontam.nozero <- filter_taxa(ps.decontam.nomitochloro, function(x) sum(x) > 0, TRUE)
  #taxa: 22313 
  #samples: 190
    
#Remove the samples I don't want to use: (13 samples removed)
    #Tag 1633 (only one sampling date; 4 additional samples)
    #Tag bonus (only one sampling date; 2 additional samples)
    #Tag 3979 (only one sampling date; 1 additional sample)
    #Tag 2300 (only one sampling date; 2 additional samples)
    #HealthState DH for coral samples (3 additional coral samples; DH ok for seawater)
        # C-105
        # Coral_18
        # C-116
 
 ps.decontam.filt_1 <- ps.decontam.nozero %>%
    subset_samples(Sample_ID !="C-105" & 
                   Sample_ID !="Coral_18" &
                   Sample_ID !="C-116" & 
                   Tag !="1633" & 
                   Tag !="bonus" & 
                   Tag !="3979" & 
                   Tag !="2300")
  #taxa: 22313 taxa
  #samples: 178
   

 Reads_per_sample_1 <- as.data.frame(sample_sums(ps.decontam.filt_1))
 mean(Reads_per_sample_1$`sample_sums(ps.decontam.filt_1)`)  #65397.13
 sd(Reads_per_sample_1$`sample_sums(ps.decontam.filt_1)`)  #64605.68
 sum(Reads_per_sample_1$`sample_sums(ps.decontam.filt_1)`) #11640690
 

```

ok well now I have 14 more samples that have >10,000 reads after removing contaminants and chloro/mito. One of those samples (Coral-13) has 9933 reads, which I'll keep. But I will removed the rest. I am going to use 9000 reads as a cut off in my code, so that I keep Coral-13. In the manuscript, I will say that I filtered at 10,000 reads, but kept Coral-13 as an exception. 
```{r}
ps.decontam.filt <- prune_samples(sample_sums(ps.decontam.filt_1) >= 9000, ps.decontam.filt_1)

Reads_per_sample <- as.data.frame(sample_sums(ps.decontam.filt))
mean(Reads_per_sample$`sample_sums(ps.decontam.filt)`)  #69736.54
sd(Reads_per_sample$`sample_sums(ps.decontam.filt)`)  #64777.55
sum(Reads_per_sample$`sample_sums(ps.decontam.filt)`) #11576265
 

write.csv(Reads_per_sample, file = "Reads_per_sample.csv")

MetaData_SamplesWeUse <- data.frame(sample_data(ps.decontam.filt))
write.csv(MetaData_SamplesWeUse, file = "MetaData_SamplesWeUse.csv")

```



#phyloseq obejct with coral and seawater from both sites
```{r}
# ps.SWandCoral.BothSites <- ps.decontam.filt %>%
#   transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
#   subset_samples(Sample_Type != "Control") %>%   #take out controls
#   subset_samples(Sample_Type != "Mock") %>% #take out mock
#   filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros
# 
# ps.ord <- ordinate(ps.SWandCoral.BothSites, "PCoA", "bray", trymax = 50, k =3, set.seed(18))
# 
# 
# BothSamp_BothSite_metadata <- data.frame(sample_data(ps.SWandCoral.BothSites))
# Perm.site <- adonis2(distance(ps.SWandCoral.BothSites, method="bray") ~ Site, data=BothSamp_BothSite_metadata, strata=BothSamp_BothSite_metadata$Tag, perm=999)
# Perm.site
# 
#     #       Number of permutations: 999
#     #                  Df SumOfSqs      R2      F Pr(>F)
#     #        Site       1    1.075 0.02255 2.8149      1
#     #        Residual 122   46.602 0.97745              
#     #        Total    123   47.678 1.00000
# 
# plot_ordination(ps.SWandCoral.BothSites, ps.ord , type="samples", color="Site", shape = "Sample_Type") +
#   geom_point(size = 4) 

```
#phyloseq obejct with Coral from both sites
```{r}
ps.coral <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Coral") %>%   #subset coral samples
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

Coral_ALL_metadata <- data.frame(sample_data(ps.coral))
perm <- how(
  blocks = Coral_ALL_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.C <- adonis2(distance(ps.coral, method="bray") ~ Site, 
                         data=Coral_ALL_metadata, 
                         perm=perm)
perm.C 

# adonis2(formula = distance(ps.coral, method = "bray") ~ Site, data = Coral_ALL_metadata, permutations = perm)
#          Df SumOfSqs      R2      F Pr(>F)
# Site      1   0.8234 0.02739 1.7458      1
# Residual 62  29.2421 0.97261              
# Total    63  30.0655 1.00000 

```
#phyloseq obejct with SW from both sites
```{r}
ps.seawater <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Seawater") %>%   #subset coral samples
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

SW_ALL_metadata <- data.frame(sample_data(ps.seawater))
perm <- how(
  blocks = SW_ALL_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.SW <- adonis2(distance(ps.seawater, method="bray") ~ Site, 
                         data=SW_ALL_metadata, 
                         perm=perm)
perm.SW 

# adonis2(formula = distance(ps.seawater, method = "bray") ~ Site, data = SW_ALL_metadata, permutations = perm)
#          Df SumOfSqs      R2      F Pr(>F)
# Site      1   1.6693 0.14629 15.251      1
# Residual 89   9.7412 0.85371              
# Total    90  11.4104 1.00000 

```





#phylo Coral, Fish Bay
```{r}
ps.coral.Fishbay <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Coral") %>% #subset coral samples
  subset_samples(Site == "Fish_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros
```

```{r}
Coral.FishBay.ord <- ordinate(ps.coral.Fishbay, "PCoA", "bray",  trymax = 100, k =3, set.seed(18))

# plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", color="Timepoint", shape = "Healthy_Diseased_Recovered") +
# #  scale_color_brewer(palette = "Oranges") +
#   geom_point(size = 4) +
#  # facet_grid(~ Tag) +
#   geom_mark_ellipse(aes(color = Healthy_Diseased_Recovered)) 
# 
# 
# plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", color="Fate", shape = "Healthy_Diseased_Recovered") +
#   geom_point(size = 4) 
# x
# 
# plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", color="Fate") +
#   geom_point(size = 4) +
#     geom_mark_ellipse(aes(filter = Fate == "Dead")) +
#   geom_mark_ellipse(aes(filter = Fate == "Diseased")) +
#   geom_mark_ellipse(aes(filter = Fate == "Healthy")) +
#   geom_mark_ellipse(aes(filter = Fate == "Recovered")) 




plot2A <- plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", color="Healthy_Diseased_Recovered") +
  geom_point(size = 4, aes(color = factor(Healthy_Diseased_Recovered))) +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Coral health state", tag = "A") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3")) 



#Because I have non-independence in my data, I need to account for the fact that my data are nested by colony tag (using block-level in permute) and within the tags/block, there is a time time series component. See:  https://uw.pressbooks.pub/appliedmultivariatestatistics/chapter/restricting-permutations/
Coral_FishBay_metadata <- data.frame(sample_data(ps.coral.Fishbay))
perm <- how(
  blocks = Coral_FishBay_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.C.FB.HDR <- adonis2(distance(ps.coral.Fishbay, method="bray") ~ Healthy_Diseased_Recovered, 
                         data=Coral_FishBay_metadata, 
                         perm=perm)
perm.C.FB.HDR 
# adonis2(formula = distance(ps.coral.Fishbay, method = "bray") ~ Healthy_Diseased_Recovered, data = Coral_FishBay_metadata, permutations = perm)
#                            Df SumOfSqs      R2      F Pr(>F)  
# Healthy_Diseased_Recovered  2   1.0707 0.05965 1.1418  0.045 *
# Residual                   36  16.8793 0.94035                
# Total                      38  17.9500 1.00000                
#this honestly isn't that different from the results if I just free permeate, so that makes me feel good about these results. 


plot2B <- plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", color="Date") +
  geom_point(size = 4) +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0, 0.5, 0.5), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Date", tag ="B")

  
#get just the legend for plotting
legend_2Date <- get_plot_component(plot2B, 'guide-box-bottom', return_all = TRUE)
legend_2Date <- as_ggplot(legend_2Date)



perm <- how(
  blocks = Coral_FishBay_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.C.FB.date <- adonis2(distance(ps.coral.Fishbay, method="bray") ~ Date, 
                          data = Coral_FishBay_metadata,
                          perm=perm)
perm.C.FB.date 
# adonis2(formula = distance(ps.coral.Fishbay, method = "bray") ~ Date, data = Coral_FishBay_metadata, permutations = perm)
#          Df SumOfSqs      R2      F Pr(>F)   
# Date     10   5.0915 0.28365 1.1087  0.005 **
# Residual 28  12.8585 0.71635                 
# Total    38  17.9500 1.00000                   
     

plot3 <- plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", shape ="Healthy_Diseased_Recovered", color = "Date") +
  facet_wrap(~Tag, scale="fixed", ncol=3)  +
  geom_point(size = 5) +
   theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical") +
        labs(color = "Date", shape = "Coral health state") + 
        scale_shape_discrete(breaks=c("Healthy", "Diseased", "Recovered"))


  


```



#phylo Seawater, Fish Bay
```{r}
ps.seawater.Fishbay <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Seawater") %>% #subset coral samples
  subset_samples(Site == "Fish_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

```

```{r}
SW.FishBay.ord <- ordinate(ps.seawater.Fishbay, "PCoA", "bray",  trymax = 200, weakties = FALSE, set.seed(18))

plot2C <- plot_ordination(ps.seawater.Fishbay, SW.FishBay.ord , type="samples", color="Healthy_Diseased_Recovered") +
  geom_point(size = 4, aes(color = factor(Healthy_Diseased_Recovered))) +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Coral health state", tag = "C") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered", "Dead"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "gray50")) 



#get just the legend for plotting
legend_2health <- get_plot_component(plot2C, 'guide-box-bottom', return_all = TRUE)
legend_2health <- as_ggplot(legend_2health)




SW_FishBay_metadata <- data.frame(sample_data(ps.seawater.Fishbay))
perm <- how(
  blocks = SW_FishBay_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.SW.FB.HDR <- adonis2(distance(ps.seawater.Fishbay, method="bray") ~ Healthy_Diseased_Recovered, 
                          data=SW_FishBay_metadata, 
                          perm=perm)
perm.SW.FB.HDR 

#                            Df SumOfSqs      R2      F Pr(>F)  
# Healthy_Diseased_Recovered  3   0.9929 0.15519 2.5718  0.035 *
# Residual                   42   5.4051 0.84481                
# Total                      45   6.3980 1.00000 



plot2D <- plot_ordination(ps.seawater.Fishbay, SW.FishBay.ord , type="samples", color="Date") +
  geom_point(size = 4) +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0, 0, 0.5), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Date", tag ="D")




perm.SW.FB.date <- adonis2(distance(ps.seawater.Fishbay, method="bray") ~ Date, 
                           data = SW_FishBay_metadata, 
                            perm=perm)
perm.SW.FB.date 
#          Df SumOfSqs      R2      F Pr(>F)   
# Date     10   3.2742 0.51175 3.6685  0.005 **
# Residual 35   3.1238 0.48825                 
# Total    45   6.3980 1.00000     



```

There are a few points in my PCoA with much higher dispersion, which is making all the other points collapse into a line. I want to make sure there is nothing wrong with my ordination by examining those points. My hunch is that the diseased water samples are just so much more different than the healthy water samples. 
```{r}
# Extract the PCoA coordinates
pcoa_coords <- SW.FishBay.ord$vectors
# Convert the PCoA coordinates to a data frame
pcoa_coords_df <- as.data.frame(pcoa_coords)
```
Ok this gives me the coordinates of each of my points. The outlier points are: 
  SW-34duplicate, SW-32, SW-29, SW-41, SW-31.
When I cross-reference these points to the stacked bar plots, I see that these same samples are the extrememly different ASV make-up samples. So I believe this is real variation and not a math issue. The healthy samples are just so much more similar to one another than the diseased samples. 



#phylo Seawater, Coral Bay
```{r}

ps.Seawater.Coralbay <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Seawater") %>% #subset Seawater samples
  subset_samples(Site == "Coral_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros



```

```{r}
SW.CoralBay.ord <- ordinate(ps.Seawater.Coralbay, "PCoA", "bray",  trymax = 50, k =2, set.seed(18))

plot1C <- plot_ordination(ps.Seawater.Coralbay, SW.CoralBay.ord , type="samples", color="Site_Disease") +
  geom_point(size = 4, aes(color = factor(Site_Disease))) +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "C") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"), 
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 
    
#get just the legend for plotting
legend_ReefHealthState <- get_plot_component(plot1C, 'guide-box-bottom', return_all = TRUE)
legend_ReefHealthState <- as_ggplot(legend_ReefHealthState)

#seawater from coral bay with HDR health state
# plot_ordination(ps.Seawater.Coralbay, SW.CoralBay.ord , type="samples", color="Healthy_Diseased_Recovered") +
#   geom_point(size = 4, aes(color = factor(Healthy_Diseased_Recovered))) +
#   theme(text=element_text(size=17),
#         legend.text=element_text(size=17),
#         panel.grid = element_blank(),
#         axis.text = element_text(size = 17),
#         legend.position = "bottom",
#         legend.direction ="vertical",
#         plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
#   guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
#   labs(color = "Coral health state", tag = "C") +
#   scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered", "Dead"),
#                        values = c("mediumseagreen", "darkorange2", "darkorchid3", "gray50"))



SW_CoralBay_metadata <- data.frame(sample_data(ps.Seawater.Coralbay))
perm <- how(
  blocks = SW_CoralBay_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.SW.CB.HDR <- adonis2(distance(ps.Seawater.Coralbay, method="bray") ~ Site_Disease, 
                          data=SW_CoralBay_metadata, 
                          perm=perm)
perm.SW.CB.HDR 

#              Df SumOfSqs      R2      F Pr(>F)   
# Site_Disease  1   0.3652 0.10925 5.2738  0.005 **
# Residual     43   2.9780 0.89075                 
# Total        44   3.3432 1.00000 


plot1D <- plot_ordination(ps.Seawater.Coralbay, SW.CoralBay.ord , type="samples", color="Date") +
  geom_point(size = 4) +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0, 0, 0.5), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Date", tag ="D")


#get just the legend for plotting
legend_Date1 <- get_plot_component(plot1D, 'guide-box-bottom', return_all = TRUE)
legend_Date1 <- as_ggplot(legend_Date1)


perm <- how(
  blocks = SW_CoralBay_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.SW.CB.date <- adonis2(distance(ps.Seawater.Coralbay, method="bray") ~ Date, 
                           data=SW_CoralBay_metadata, 
                          perm=perm)
perm.SW.CB.date 

#          Df SumOfSqs      R2      F Pr(>F)   
# Date     11   2.8474 0.85168 17.227  0.005 **
# Residual 33   0.4959 0.14832                 
# Total    44   3.3432 1.00000  

```

checking <- as.data.frame(rownames(otus))
checking$ASVsum <- rowSums(otus)
hist(checking$ASVsum)

SW_CoralBay_OTU <- as.data.frame(otu_table(ps.Seawater.Coralbay))
check_CB_SW <- as.data.frame(rownames(SW_CoralBay_OTU))
check_CB_SW$ASVsum <- rowSums(SW_CoralBay_OTU)
hist(check_CB_SW$ASVsum)


#phylo Coral, Coral Bay
```{r}

ps.Coral.Coralbay <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Coral") %>% #subset Seawater samples
  subset_samples(Site == "Coral_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros


```

```{r}
Coral.CoralBay.ord <- ordinate(ps.Coral.Coralbay, "PCoA", "bray",  trymax = 50, k =2, set.seed(18))



plot1A <- plot_ordination(ps.Coral.Coralbay, Coral.CoralBay.ord , type="samples", color="Site_Disease") +
  geom_point(size = 4, aes(color = factor(Site_Disease))) +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "A") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"),
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 


#coral from coral bay with HDR health state
 # plot_ordination(ps.Coral.Coralbay, Coral.CoralBay.ord , type="samples", color="Healthy_Diseased_Recovered") +
 #  geom_point(size = 4, aes(color = factor(Healthy_Diseased_Recovered))) +
 #  theme(text=element_text(size=17),
 #        legend.text=element_text(size=17),
 #        panel.grid = element_blank(),
 #        axis.text = element_text(size = 17),
 #        legend.position = "bottom",
 #        legend.direction ="vertical",
 #        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
 #  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
 #  labs(color = "Coral health state", tag = "C") +
 #  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered", "Dead"),
 #                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "gray50"))



Coral_CoralBay_metadata <- data.frame(sample_data(ps.Coral.Coralbay))
perm <- how(
  blocks = Coral_CoralBay_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.C.CB.HDR <- adonis2(distance(ps.Coral.Coralbay, method="bray") ~ Site_Disease, 
                         data=Coral_CoralBay_metadata,
                         perm=perm)
perm.C.CB.HDR 
# 
#              Df SumOfSqs      R2     F Pr(>F)  
# Site_Disease  1   0.5843 0.05174 1.255  0.045 *
# Residual     23  10.7078 0.94826               
# Total        24  11.2921 1.00000 




plot1B <- plot_ordination(ps.Coral.Coralbay, Coral.CoralBay.ord , type="samples", color="Date") +
  geom_point(size = 4) +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0, 0.5, 0.5), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Date", tag = "B")

perm.C.CB.date <- adonis2(distance(ps.Coral.Coralbay, method="bray") ~ Date, 
                          data = Coral_CoralBay_metadata, 
                          perm=perm)
perm.C.CB.date 

#          Df SumOfSqs      R2      F Pr(>F)   
# Date      9   4.6194 0.40908 1.1538  0.005 **
# Residual 15   6.6727 0.59092                 
# Total    24  11.2921 1.00000      

```




#put plots together for Fig 1
```{r}

plot_grid(plot1A + theme(legend.position = "none"), 
          plot1B + theme(legend.position = "none"), 
          plot1C + theme(legend.position = "none"), 
          plot1D + theme(legend.position = "none"), 
          legend_ReefHealthState, 
          legend_Date1, 
          ncol = 2, nrow = 3)

```





#put plots together for Fig 2
```{r}

plot_grid(plot2A + theme(legend.position = "none"), 
          plot2B + theme(legend.position = "none"), 
          plot2C + theme(legend.position = "none"), 
          plot2D + theme(legend.position = "none"), 
          legend_2health, 
          legend_2Date, 
          ncol = 2, nrow = 3)

```


#SCTLD Prevalence
```{r}
Prevelance_ForAnalyses <- read.csv("~/Documents/WHOI/Projects/STJ_SCTLD_TimeseriesExperiment/Disease Wheel Data/Prevelance_ForAnalyses.csv")
Prevelance_ForAnalyses$Date_nice <- as.Date(Prevelance_ForAnalyses$Date) #make ggplot know it's a date


ggplot()+
  geom_point(data=Prevelance_ForAnalyses, aes(x = Date_nice, y=Prev_Percent), na.rm = TRUE, color="Red") +
  geom_smooth(data=Prevelance_ForAnalyses, aes(x = Date_nice, y=Prev_Percent), na.rm = TRUE, method = loess, color="Red", se = FALSE) +
  scale_y_continuous(name = "SCTLD Prevelance (percent)") +
  xlab("Date") +
  theme(text=element_text(size=13))+
  ggtitle("SCTLD Prevelance in Fish Bay Seawater") 

#For Loess:
#default span is α = 0.75
```





#Beta Diversity 
MODIFIED FROM CYNTHIA: Code to Reproduce R Figures
I want to double check the date vs date_nice effect -- 9/13/24

useful resource on beta div types: https://docs.onecodex.com/en/articles/4150649-beta-diversity
Code modified from Anya from Cleanerfish project and LCT project

## Means: SW from Fish Bay
I am trying to find the mean beta diversity for either each time point, or for the epidemic vs endemic phases

```{r}
#using: ps.seawater.Fishbay

otu_rel <- as.data.frame(otu_table(ps.seawater.Fishbay)) #pull out OTU table
  otu_rel[1:3,1:3] #make sure sample names are rows
metadat_rel <- as.data.frame(sample_data(ps.seawater.Fishbay)) #pull out Meta table

#Check if my rows are in the same order
identical(rownames(otu_rel), rownames(metadat_rel))
    #True!

#make a distance matrix, and specify the method - in this case, bray curtis 
# see https://rdrr.io/cran/vegan/man/vegdist.html
# how dissimilar is each sample is from every other sample
Veg <- vegdist(otu_rel, method="bray") 

# model that that we use to test the differences in distance to the centroid within our groups in the column Treatment
# takes the matrix and figues out how dispersed the samples are within groups
# Quantifies the dispersion between each of the points 
# How dispersed are the groups -- Find the centroid then report how far each point is from the centroid
Mod_BetaDisp <-betadisper(Veg, metadat_rel$Healthy_Diseased_Recovered)

Mod_BetaDisp
summary(Mod_BetaDisp)

Mod_aov <- anova(Mod_BetaDisp)
Mod_aov
    # Response: Distances
    #           Df  Sum Sq  Mean Sq F value    Pr(>F)    
    # Groups     3 0.46078 0.153595  7.4748 0.0004052 ***
    # Residuals 42 0.86304 0.020548


##################################################

##Prepare for plotting 

dist.tab_SW_FB <- as.data.frame(Mod_BetaDisp$distances)
dist.tab_SW_FB$Sample_names <- rownames(dist.tab_SW_FB)
colnames(dist.tab_SW_FB)[1] <- "distance"

identical(rownames(metadat_rel), dist.tab_SW_FB$Sample_names)
    #TRUE

dist.tab_SW_FB$Healthy_Diseased_Recovered <- metadat_rel$Healthy_Diseased_Recovered
dist.tab_SW_FB$Date <- metadat_rel$Date
dist.tab_SW_FB$Disease_Phase <- metadat_rel$Disease_Phase
dist.tab_SW_FB

######### Code from: 
#   http://www.sthda.com/english/wiki/ggplot2-error-bars-quick-start-guide-r-software-and-data-visualization#google_vignette

# Function to calculate the mean and the standard deviation for each group
data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}

#Summarize the data :
df2 <- data_summary(dist.tab_SW_FB, varname="distance", 
                    groupnames=c("Healthy_Diseased_Recovered", "Disease_Phase"))

# I am going to add a date that's in the middle of each phase. This is arbitrary and for plotting purposes. I will add  notation in Illustrator that the mean is across the entire disease phase. I am mak
df2$Date_forPlot <- c("2023-06-01", #Endemic
                      "2021-06-30", #Epidemic
                      "2023-06-01", #Endemic
                      "2021-06-01", #Epidemic
                      "2020-07-16", #Disease_discovered
                      "2021-06-01", #Epidemic
                      "2023-06-01", #Endemic
                      "2021-05-01" #Epidemic
                      )
df2$Date_forPlot <- as.Date(df2$Date_forPlot)

ggplot() + 
 geom_point(data=Prevelance_ForAnalyses, aes(x = Date_nice, y=Prev_Percent/10), na.rm = TRUE, color="Red", size=2.4) +
  geom_smooth(data=Prevelance_ForAnalyses, aes(x = Date_nice, y=Prev_Percent/10), na.rm = TRUE, method = loess, color="Red", se = FALSE) +
  xlab("Date") +
  theme(text=element_text(size=13)) +
  scale_y_continuous(name = "Distance to Centroid (colored shapes)",
    sec.axis = sec_axis(~.*10, name="Percent SCTLD Prevelance (red line and red points)")) +
geom_point(data = df2, aes(x=Date_forPlot, y=distance, color=Healthy_Diseased_Recovered, shape=Disease_Phase), size=5) +
  geom_errorbar(data = df2, aes(x=Date_forPlot, y=distance, ymin=distance-sd, ymax=distance+sd, color=Healthy_Diseased_Recovered), width=50, size=0.75,) +
  theme_classic()+
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered", "Dead"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "gray50")) 

rm(otu_rel)
rm(metadat_rel)
rm(Veg)
rm(Mod_BetaDisp)
rm(Mod_aov)
rm(data_summary)
rm(df2)
```




##DispXPrev - Fish Bay Seawater
```{r}
metadat_rel <- as.data.frame(as.matrix(sample_data(ps.seawater.Fishbay)))

otu_rel <- as.matrix(otu_table(ps.seawater.Fishbay))
otu_rel2 <- as.data.frame(otu_rel)
otu_rel2 <- otu_rel2[, colSums(otu_rel2 != 0) > 0] #get rid of ASVs that have no abundance in any samples
#otu_rel_tra <- t(otu_rel2) #make sure sample names are rows
ASV.coral.log <- log(otu_rel2+1) #Log transform the realtive abundance data, where I add a 1 pseudocount to everything

ASV.coral.log.b <- vegdist(ASV.coral.log, "bray") #calculate bray curtis dissimilarity

#calculate multivariate dispersions based on condition - Distances to the centroid!
mod2 <-betadisper(ASV.coral.log.b, metadat_rel$Date)

#Get the distances to centroid from the model
dis2 <- as.data.frame(mod2$distances) #save the distances
dis2.meta <- cbind(dis2, metadat_rel) #combine distances with the metadata
colnames(dis2.meta)[1] <- "distance"

dis2.meta$Date_nice <- as.Date(dis2.meta$Date) #make ggplot know it's a date

#run linear model and Kruskal-Wallis test to test significance
distlm2 <-lm(distance~Date, data=dis2.meta)
anova(distlm2)
      # Response: distance
      #           Df  Sum Sq  Mean Sq F value Pr(>F)
      # Date      10 0.50303 0.050303   1.222 0.3114
      # Residuals 35 1.44073 0.041164  

summary(distlm2)
      # Residual standard error: 0.2029 on 35 degrees of freedom
      # Multiple R-squared:  0.2588,	Adjusted R-squared:  0.04702 
      # F-statistic: 1.222 on 10 and 35 DF,  p-value: 0.3114


Prevelance_ForAnalyses <- read.csv("~/Documents/WHOI/Projects/STJ_SCTLD_TimeseriesExperiment/Disease Wheel Data/Prevelance_ForAnalyses.csv")
Prevelance_ForAnalyses$Date_nice <- as.Date(Prevelance_ForAnalyses$Date) #make ggplot know it's a date


plot <- ggplot()+
  geom_point(data = dis2.meta, aes(x = Date, y=distance), color="Blue") +
  geom_smooth(data = dis2.meta, aes(x = Date, y=distance), method = "lm", color="Blue", se = FALSE) +
  geom_point(data=Prevelance_ForAnalyses, aes(x = Date_nice, y=Prev_Percent/5), na.rm = TRUE, color="Red") +
  geom_smooth(data=Prevelance_ForAnalyses, aes(x = Date_nice, y=Prev_Percent/5), na.rm = TRUE, method = "lm", color="Red", se = FALSE) +
  scale_y_continuous(name = "Distance to Centroid",
    sec.axis = sec_axis(~.*5, name="SCTLD Prevelance (percent)")) +
  xlab("Date") +
  theme(text=element_text(size=13))+
  ggtitle("Beta diversity and SCTLD Prevelance in Fish Bay Seawater") +
  theme(axis.title.y = element_text(color = "Blue", size=13),
    axis.title.y.right = element_text(color = "Red", size=13))

plot

rm(metadat_rel)
rm(otu_rel)
rm(otu_rel2)
rm(ASV.coral.log)
rm(ASV.coral.log.b)
rm(mod2)
rm(dis2)
rm(dis2.meta)
rm(distlm2)
```

##DispXPrev - add Fish Bay Coral
```{r}
metadat_rel <- as.data.frame(as.matrix(sample_data(ps.coral.Fishbay)))

otu_rel <- as.matrix(otu_table(ps.coral.Fishbay))
otu_rel2 <- as.data.frame(otu_rel)
otu_rel2 <- otu_rel2[, colSums(otu_rel2 != 0) > 0] #get rid of ASVs that have no abundance in any samples
#otu_rel_tra <- t(otu_rel2) #make sure sample names are rows
ASV.coral.log <- log(otu_rel2+1) #Log transform the realtive abundance data, where I add a 1 pseudocount to everything

ASV.coral.log.b <- vegdist(ASV.coral.log, "bray") #calculate bray curtis dissimilarity

#calculate multivariate dispersions based on condition - Distances to the centroid!
mod2 <-betadisper(ASV.coral.log.b, metadat_rel$Date)

#Get the distances to centroid from the model
dis2 <- as.data.frame(mod2$distances) #save the distances
dis2.meta <- cbind(dis2, metadat_rel) #combine distances with the metadata
colnames(dis2.meta)[1] <- "distance"

dis2.meta$Date_nice <- as.Date(dis2.meta$Date)

#run linear model and Kruskal-Wallis test to test significance
distlm2 <-lm(distance~Date, data=dis2.meta)
anova(distlm2)
      # Response: distance
      #           Df   Sum Sq   Mean Sq F value  Pr(>F)  
      # Date      10 0.065343 0.0065343  6.5882 2.617e-05 ***
      # Residuals 30 0.029755 0.0009918                   
     
summary(distlm2)
      # Residual standard error: 0.04635 on 39 degrees of freedom
      # Multiple R-squared:  0.6871,	Adjusted R-squared:  0.5828 
      # F-statistic: 6.588 on 10 and 30 DF,  p-value: 2.617e-05


plot + 
  geom_point(data = dis2.meta, aes(x = Date_nice, y=distance), color="purple") +
  geom_smooth(data = dis2.meta, aes(x = Date_nice, y=distance), method = "lm", color="purple", se = FALSE) 



rm(metadat_rel)
rm(otu_rel)
rm(otu_rel2)
rm(ASV.coral.log)
rm(ASV.coral.log.b)
rm(mod2)
rm(dis2)
rm(dis2.meta)
rm(distlm2)
```

##DispXPrev - Coral Bay Seawater
```{r}
metadat_rel <- as.data.frame(as.matrix(sample_data(ps.Seawater.Coralbay)))

otu_rel <- as.matrix(otu_table(ps.Seawater.Coralbay))
otu_rel2 <- as.data.frame(otu_rel)
otu_rel2 <- otu_rel2[, colSums(otu_rel2 != 0) > 0] #get rid of ASVs that have no abundance in any samples
#otu_rel_tra <- t(otu_rel2) #make sure sample names are rows
ASV.coral.log <- log(otu_rel2+1) #Log transform the realtive abundance data, where I add a 1 pseudocount to everything

ASV.coral.log.b <- vegdist(ASV.coral.log, "bray") #calculate bray curtis dissimilarity

#calculate multivariate dispersions based on condition - Distances to the centroid!
mod2 <-betadisper(ASV.coral.log.b, metadat_rel$Date)

#Get the distances to centroid from the model
dis2 <- as.data.frame(mod2$distances) #save the distances
dis2.meta <- cbind(dis2, metadat_rel) #combine distances with the metadata
colnames(dis2.meta)[1] <- "distance"

dis2.meta$Date_nice <- as.Date(dis2.meta$Date)

#run linear model and Kruskal-Wallis test to test significance
distlm2 <-lm(distance~Date, data=dis2.meta)
anova(distlm2)
      #Response: distance
      #          Df   Sum Sq   Mean Sq F value    Pr(>F)    
      # Date_nice  1 0.000179 0.00017927   0.068 0.7955
      # Residuals 43 0.113374 0.00263660 
summary(distlm2)
      # Residual standard error: 0.05135 on 43 degrees of freedom
      # Multiple R-squared:  0.001579,	Adjusted R-squared:  -0.02164 
      # F-statistic: 0.06799 on 1 and 43 DF,  p-value: 0.7955



dis2.meta$Prev_Percent <- as.numeric(dis2.meta$Prev_Percent)

plot <- ggplot(dis2.meta)+
  geom_point(aes(x = Date_nice, y=distance), color="Blue") +
  geom_smooth(aes(x = Date_nice, y=distance), method = "lm", color="Blue", se = FALSE) +
  geom_point(data=subset(dis2.meta, !is.na(Prev_Percent)), aes(x = Date_nice, y=Prev_Percent/10), na.rm = TRUE, color="Red") +
  geom_smooth(data=subset(dis2.meta, !is.na(Prev_Percent)), aes(x = Date_nice, y=Prev_Percent/10), na.rm = TRUE, method = "lm", color="Red", se = FALSE) +
  scale_y_continuous(name = "Distance to Centroid",
    sec.axis = sec_axis(~.*10, name="SCTLD Prevelance (percent)")) +
  xlab("Date") +
  theme(text=element_text(size=13))+
  ggtitle("Beta div.and SCTLD Prev. in CoralBay") +
  theme(axis.title.y = element_text(color = "Blue", size=13),
    axis.title.y.right = element_text(color = "Red", size=13))

plot

rm(metadat_rel)
rm(otu_rel)
rm(otu_rel2)
rm(ASV.coral.log)
rm(ASV.coral.log.b)
rm(mod2)
rm(dis2)
rm(dis2.meta)
rm(distlm2)
```

##DispXPrev - add Coral Bay Coral
```{r}
metadat_rel <- as.data.frame(as.matrix(sample_data(ps.Coral.Coralbay)))

otu_rel <- as.matrix(otu_table(ps.Coral.Coralbay))
otu_rel2 <- as.data.frame(otu_rel)
otu_rel2 <- otu_rel2[, colSums(otu_rel2 != 0) > 0] #get rid of ASVs that have no abundance in any samples
#otu_rel_tra <- t(otu_rel2) #make sure sample names are rows
ASV.coral.log <- log(otu_rel2+1) #Log transform the realtive abundance data, where I add a 1 pseudocount to everything

ASV.coral.log.b <- vegdist(ASV.coral.log, "bray") #calculate bray curtis dissimilarity

#calculate multivariate dispersions based on condition - Distances to the centroid!
mod2 <-betadisper(ASV.coral.log.b, metadat_rel$Date)

#Get the distances to centroid from the model
dis2 <- as.data.frame(mod2$distances) #save the distances
dis2.meta <- cbind(dis2, metadat_rel) #combine distances with the metadata
colnames(dis2.meta)[1] <- "distance"

dis2.meta$Date_nice <- as.Date(dis2.meta$Date)

#run linear model and Kruskal-Wallis test to test significance
distlm2 <-lm(distance~Date, data=dis2.meta)
anova(distlm2)
      #Response: distance
      #          Df   Sum Sq   Mean Sq F value    Pr(>F)    
      #Date      6 0.018014 0.0030023  0.4214 0.8571
      #Residuals 23 0.163881 0.0071253     
summary(distlm2)
  #        Residual standard error: 0.08441 on 23 degrees of freedom
  #        Multiple R-squared:  0.09903,	Adjusted R-squared:  -0.136 
  #        F-statistic: 0.4214 on 6 and 23 DF,  p-value: 0.8571


Prev_data_noNA<- na.omit(dis2.meta)

dis2.meta$Prev_Percent <- as.numeric(dis2.meta$Prev_Percent)

plot + 
  geom_point(data = dis2.meta, aes(x = Date_nice, y=distance), color="purple") +
  geom_smooth(data = dis2.meta, aes(x = Date_nice, y=distance), method = "lm", color="purple", se = FALSE) 

rm(metadat_rel)
rm(otu_rel)
rm(otu_rel2)
rm(ASV.coral.log)
rm(ASV.coral.log.b)
rm(mod2)
rm(dis2)
rm(dis2.meta)
rm(distlm2)

```




#Stacked Bar plots

##rel abun >= 1% for Seawater from Fish Bay
- code is almost entirely from Chapt GPT
- at first, I tried plotting the ASVs that make up 90% cumulative relative abundance for each sample. But that gave me a total of 1175 unique ASVs, which is more than the number of colors in R, and would not be useful for plotitng.
```{r}
abundance_table <- as.data.frame(otu_table(ps.seawater.Fishbay))
  #46 obs of 7023 variables
abundance_table <- abundance_table[, colSums(abundance_table != 0) > 0]
  #46 obs of 7023 variables

meta_SW.FishBay <- as.data.frame(sample_data(ps.seawater.Fishbay))
data_frame_merge <- merge(abundance_table, meta_SW.FishBay,
                          by = 'row.names', all = TRUE)

# Melt the abundance table to long format for ggplot2
abundance_long <- data_frame_merge %>%
  pivot_longer(cols=starts_with("ASV"), 
               names_to = "ASV_ID", 
               values_to = "Abundance") %>%
  arrange(Row.names, desc(Abundance))
dim(abundance_long) #323058     27

# Filter ASVs with relative abundance >= 0.01 (1%)
abundance_long <- abundance_long %>%
  filter(Abundance >= 0.01)
dim(abundance_long) #858  27 --> lovely!

#how many ASVs are left to plot?
length(unique(abundance_long$ASV_ID)) #114
n <- 114 #plug that number in here to a get that number of unique colors to plot
palette <- distinctColorPalette(n)
  
# Create the stacked bar plot
ggplot(abundance_long, aes(x = Date, y = Abundance, fill = ASV_ID)) +
  geom_bar(stat = "identity", position="stack") +
  facet_grid(~Tag, scale="free") +
 # theme_minimal() +
  scale_fill_manual(values=palette) +
  labs(y = "Relative Abundance", fill = "ASV") +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, hjust = 1),
        legend.position = "none") +
  labs(title = "ASVs with at least 1% relative abundance from Seawater from Fish Bay") +
  xlab(" ")


rm(abundance_table)
rm(meta_SW.FishBay)
rm(data_frame_merge)
rm(abundance_long)
rm(n)
rm(palette)
```


##rel abun >= 1% for Corals from Fish Bay
```{r}
abundance_table <- as.data.frame(otu_table(ps.coral.Fishbay))
  #41 obs of 3871 variables
abundance_table <- abundance_table[, colSums(abundance_table != 0) > 0]
  #41 obs of 3871 variables

meta_Coral.FishBay <- as.data.frame(sample_data(ps.coral.Fishbay))
data_frame_merge <- merge(abundance_table, meta_Coral.FishBay,
                          by = 'row.names', all = TRUE)

# Melt the abundance table to long format for ggplot2
abundance_long <- data_frame_merge %>%
  pivot_longer(cols=starts_with("ASV"), 
               names_to = "ASV_ID", 
               values_to = "Abundance") %>%
  arrange(Row.names, desc(Abundance))
dim(abundance_long) #158711     27

# Filter ASVs with relative abundance >= 0.01 (1%)
abundance_long <- abundance_long %>%
  filter(Abundance >= 0.01)
dim(abundance_long) #687  27 --> lovely!

#how many ASVs are left to plot?
length(unique(abundance_long$ASV_ID)) #544
n <- 544 #plug that number in here to a get that number of unique colors to plot
palette <- distinctColorPalette(n)
  
# Create the stacked bar plot
ggplot(abundance_long, aes(x = Date, y = Abundance, fill = ASV_ID)) +
  geom_bar(stat = "identity", position="stack") +
  facet_grid(~Tag, scale="free") +
 # theme_minimal() +
  scale_fill_manual(values=palette) +
  labs(y = "Relative Abundance", fill = "ASV") +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, hjust = 1),
        legend.position = "none") +
  labs(title = "ASVs with at least 1% relative abundance from Coral from Fish Bay") +
  xlab(" ")


rm(abundance_table)
rm(meta_Coral.FishBay)
rm(data_frame_merge)
rm(abundance_long)
rm(n)
rm(palette)
```


##rel abun >= 1% for Corals from Coral Bay
```{r}
abundance_table <- as.data.frame(otu_table(ps.Coral.Coralbay))
  #35 obs of 5140 variables
abundance_table <- abundance_table[, colSums(abundance_table != 0) > 0]
  #35 obs of 5140 variables

meta_Coral.CoralBay <- as.data.frame(sample_data(ps.Coral.Coralbay))
data_frame_merge <- merge(abundance_table, meta_Coral.CoralBay,
                          by = 'row.names', all = TRUE)

# Melt the abundance table to long format for ggplot2
abundance_long <- data_frame_merge %>%
  pivot_longer(cols=starts_with("ASV"), 
               names_to = "ASV_ID", 
               values_to = "Abundance") %>%
  arrange(Row.names, desc(Abundance))
dim(abundance_long) #179900     27

# Filter ASVs with relative abundance >= 0.01 (1%)
abundance_long <- abundance_long %>%
  filter(Abundance >= 0.01)
dim(abundance_long) #525  27 --> lovely!

#how many ASVs are left to plot?
length(unique(abundance_long$ASV_ID)) #459
n <- 459 #plug that number in here to a get that number of unique colors to plot
palette <- distinctColorPalette(n)
  
# Create the stacked bar plot
ggplot(abundance_long, aes(x = Date, y = Abundance, fill = ASV_ID)) +
  geom_bar(stat = "identity", position="stack") +
  facet_grid(~Tag, scale="free") +
 # theme_minimal() +
  scale_fill_manual(values=palette) +
  labs(y = "Relative Abundance", fill = "ASV") +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, hjust = 1),
        legend.position = "none") +
  labs(title = "ASVs with at least 1% relative abundance from Coral from Coral Bay") +
  xlab(" ")


rm(abundance_table)
rm(meta_Coral.CoralBay)
rm(data_frame_merge)
rm(abundance_long)
rm(n)
rm(palette)
```



##rel abun >= 1% for Seawater from Coral Bay
```{r}
abundance_table <- as.data.frame(otu_table(ps.Seawater.Coralbay))
  #45 obs of 9158 variables
abundance_table <- abundance_table[, colSums(abundance_table != 0) > 0]
  #45 obs of 9158 

meta_Seawater.CoralBay <- as.data.frame(sample_data(ps.Seawater.Coralbay))
data_frame_merge <- merge(abundance_table, meta_Seawater.CoralBay,
                          by = 'row.names', all = TRUE)

# Melt the abundance table to long format for ggplot2
abundance_long <- data_frame_merge %>%
  pivot_longer(cols=starts_with("ASV"), 
               names_to = "ASV_ID", 
               values_to = "Abundance") %>%
  arrange(Row.names, desc(Abundance))
dim(abundance_long) #412110     27

# Filter ASVs with relative abundance >= 0.01 (1%)
abundance_long <- abundance_long %>%
  filter(Abundance >= 0.01)
dim(abundance_long) #859  27 --> lovely!

#how many ASVs are left to plot?
length(unique(abundance_long$ASV_ID)) #73
n <- 73 #plug that number in here to a get that number of unique colors to plot
palette <- distinctColorPalette(n)
  
# Create the stacked bar plot
ggplot(abundance_long, aes(x = Date, y = Abundance, fill = ASV_ID)) +
  geom_bar(stat = "identity", position="stack") +
  facet_grid(~Tag, scale="free") +
 # theme_minimal() +
  scale_fill_manual(values=palette) +
  labs(y = "Relative Abundance", fill = "ASV") +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, hjust = 1),
        legend.position = "none") +
  labs(title = "ASVs with at least 1% relative abundance from Seawater from Coral Bay") +
  xlab(" ")


rm(abundance_table)
rm(meta_Seawater.CoralBay)
rm(data_frame_merge)
rm(abundance_long)
rm(n)
rm(palette)
```




##20 most abundant AVS -- Corals from Fish Bay
```{r}
asv_abundance <- as.data.frame(as.matrix(otu_table(ps.coral.Fishbay)))
  #41 obs of 3873 variables
asv_abundance <- asv_abundance[, colSums(asv_abundance != 0) > 0]
  #41 obs of 3873 variables 

# Calculate the total abundance of ASVs per sample
sample_total_abundance <- colSums(asv_abundance)

# Sort the samples by total abundance in descending order
sorted_samples <- sample_total_abundance[order(sample_total_abundance, decreasing = TRUE)]
head(sorted_samples)
top_20_asvs <- names(sorted_samples[1:20])

# Get the top 20 most abundant ASVs
top_20_asvs_samples <- asv_abundance[, top_20_asvs]
top_20_asvs_samples$Sample_names <- row.names(top_20_asvs_samples)
```

```{r}
# Get unique values in the specified category
meta_top20 <- META[top_20_asvs_samples$Sample_names]

taxa_top20 <- as.data.frame(taxa[top_20_asvs,])
taxa_top20$ASV_ID <- row.names(taxa_top20)
taxa_top20 <- taxa_top20 %>% arrange(ASV_ID)

write.csv(taxa_top20, "Top20_ASV_FishBay_Corals.csv")
 


#Combine the data frames into one
data_frame_merge <- merge(meta_top20, top_20_asvs_samples,
                          by = 'row.names', all = TRUE)

#Reshape the data into long format
long_df_FB_coral <- data_frame_merge %>% pivot_longer(
                    cols=starts_with("ASV"),
                    names_to='ASV_ID',
                    values_to='ASV_prop') %>% 
                    arrange(ASV_ID)

mean_per_sample_FB <- long_df_FB_coral %>% 
                  group_by(Sample_ID) %>%
                  summarise(mean = mean(ASV_prop),
                            SD = sd(ASV_prop),
                            sum = sum(ASV_prop))
    # min sum =  0.003954271, max sum = 0.847322191

total_mean_FB <- mean_per_sample_FB %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.2762619, SD = 0.2591617



palette20 <- c("#C5EC8A",  "#6EB3EB",  "#4CEADD", "#ECF143", "#C571D9", "#962BEF"  , "#DFA643",    "#E64A8A", "#55AE62", "#E45F4E", "#E17FAD", "#61E97D", "#E9D2A9", "#6B6E7F", "#E33EE6", "#ACAA51", "#A5D4E8",  "#BC9DE1",  "#6765F0", "#B99299")





ggplot(long_df_FB_coral, aes(x=Date, y=ASV_prop, fill=ASV_ID)) + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Tag, scale="free") +
  scale_fill_manual(values=palette20) +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  labs(title = "Top 20 ASVs from Corals from Fish Bay") +
  xlab(" ")
#%#%#%#%


```
Top20ASVsfromCoralsfromFishBay






##20 most abundant AVS  -- Seawater from Fish Bay
```{r}
asv_abundance <- as.data.frame(as.matrix(otu_table(ps.seawater.Fishbay)))
  #46 obs of 7023 variables
asv_abundance <- asv_abundance[, colSums(asv_abundance != 0) > 0]
  #46 obs of 7023 variables 
  
# Calculate the total abundance of ASVs per sample
sample_total_abundance <- colSums(asv_abundance)

# Sort the samples by total abundance in descending order
sorted_samples <- sample_total_abundance[order(sample_total_abundance, decreasing = TRUE)]
head(sorted_samples)
top_20_asvs <- names(sorted_samples[1:20])

# Get the top 20 most abundant ASVs
top_20_asvs_samples <- asv_abundance[, top_20_asvs]
top_20_asvs_samples$Sample_names <- row.names(top_20_asvs_samples)
```

```{r}
# Get unique values in the specified category
meta_top20 <- META[top_20_asvs_samples$Sample_names]

taxa_top20 <- as.data.frame(taxa[top_20_asvs,])
taxa_top20$ASV_ID <- row.names(taxa_top20)
taxa_top20 <- taxa_top20 %>% arrange(ASV_ID)

write.csv(taxa_top20, "Top20_ASV_FishBay_SW.csv")
 


#Combine the data frames into one
data_frame_merge <- merge(meta_top20, top_20_asvs_samples,
                          by = 'row.names', all = TRUE)

write.csv(data_frame_merge, "Top20_Short_FB_SW.csv")

#Reshape the data into long format
long_df_FB_SW <- data_frame_merge %>% pivot_longer(
                    cols=starts_with("ASV"),
                    names_to='ASV_ID',
                    values_to='ASV_prop') %>% 
                    arrange(ASV_ID)


mean_per_sample_FB_SW <- long_df_FB_SW %>% 
                  group_by(Sample_ID) %>%
                  summarise(mean = mean(ASV_prop),
                            SD = sd(ASV_prop),
                            sum = sum(ASV_prop))
    # min sum =  0.2414640, max sum = 0.6264272

total_mean_FB_SW <- mean_per_sample_FB_SW %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.5018402, SD = 0.07697495




palette20 <- c("#C5EC8A",  "#6EB3EB",  "#4CEADD", "#ECF143", "#C571D9", "#962BEF"  , "#DFA643",    "#E64A8A", "#55AE62", "#E45F4E", "#E17FAD", "#61E97D", "#E9D2A9", "#6B6E7F", "#E33EE6", "#ACAA51", "#A5D4E8",  "#BC9DE1",  "#6765F0", "#B99299")





ggplot(long_df_FB_SW, aes(x=Date, y=ASV_prop, fill=ASV_ID)) + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Tag, scale="free") +
  scale_fill_manual(values=palette20) +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  labs(title = "Top 20 ASVs from Seawater from Fish Bay") +
  xlab(" ")
#%#%#%#%


```
Top20ASVsfromSWfromFishBay







##20 most abundant AVS  -- Coral from Coral Bay
```{r}
asv_abundance <- as.data.frame(as.matrix(otu_table(ps.Coral.Coralbay)))
  #35 obs of 5140 variables
asv_abundance <- asv_abundance[, colSums(asv_abundance != 0) > 0]
  #35 obs of 5140 variables 
  
# Calculate the total abundance of ASVs per sample
sample_total_abundance <- colSums(asv_abundance)

# Sort the samples by total abundance in descending order
sorted_samples <- sample_total_abundance[order(sample_total_abundance, decreasing = TRUE)]
head(sorted_samples)
top_20_asvs <- names(sorted_samples[1:20])

# Get the top 20 most abundant ASVs
top_20_asvs_samples <- asv_abundance[, top_20_asvs]
top_20_asvs_samples$Sample_names <- row.names(top_20_asvs_samples)
```

```{r}
# Get unique values in the specified category
meta_top20 <- META[top_20_asvs_samples$Sample_names]

taxa_top20 <- as.data.frame(taxa[top_20_asvs,])
taxa_top20$ASV_ID <- row.names(taxa_top20)
taxa_top20 <- taxa_top20 %>% arrange(ASV_ID)

write.csv(taxa_top20, "Top20_ASV_CoralBay_Coral.csv")
 


#Combine the data frames into one
data_frame_merge <- merge(meta_top20, top_20_asvs_samples,
                          by = 'row.names', all = TRUE)

#Reshape the data into long format
long_df_CB_coral <- data_frame_merge %>% pivot_longer(
                    cols=starts_with("ASV"),
                    names_to='ASV_ID',
                    values_to='ASV_prop') %>% 
                    arrange(ASV_ID)


mean_per_sample_CB <- long_df_CB_coral %>% 
                  group_by(Sample_ID) %>%
                  summarise(mean = mean(ASV_prop),
                            SD = sd(ASV_prop),
                            sum = sum(ASV_prop))
    # min sum =  0.000000000, max sum = 0.771292005

total_mean_CB <- mean_per_sample_CB %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.2271098, SD = 0.2465084


mean_per_sample_CBandFB <- rbind(mean_per_sample_CB, mean_per_sample_FB)

#mean for coral
total_mean_CB_and_FB <- mean_per_sample_CBandFB %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.2536261, SD = 0.2529365 


palette20 <- c("#C5EC8A",  "#6EB3EB",  "#4CEADD", "#ECF143", "#C571D9", "#962BEF"  , "#DFA643",    "#E64A8A", "#55AE62", "#E45F4E", "#E17FAD", "#61E97D", "#E9D2A9", "#6B6E7F", "#E33EE6", "#ACAA51", "#A5D4E8",  "#BC9DE1",  "#6765F0", "#B99299")





ggplot(long_df_CB_coral, aes(x=Date, y=ASV_prop, fill=ASV_ID)) + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Tag, scale="free") +
  scale_fill_manual(values=palette20) +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  labs(title = "Top 20 ASVs from Coral from Coral Bay") +
  xlab(" ")
#%#%#%#%


```
Top20ASVsfromCoralfromCoralBay











##20 most abundant AVS  -- Seawater from Coral Bay
```{r}
asv_abundance <- as.data.frame(as.matrix(otu_table(ps.Seawater.Coralbay)))
  #45 obs of 9158 variables
asv_abundance <- asv_abundance[, colSums(asv_abundance != 0) > 0]
  #45 obs of 9158 variables 
  
# Calculate the total abundance of ASVs per sample
sample_total_abundance <- colSums(asv_abundance)

# Sort the samples by total abundance in descending order
sorted_samples <- sample_total_abundance[order(sample_total_abundance, decreasing = TRUE)]
head(sorted_samples)
top_20_asvs <- names(sorted_samples[1:20])

# Get the top 20 most abundant ASVs
top_20_asvs_samples <- asv_abundance[, top_20_asvs]
top_20_asvs_samples$Sample_names <- row.names(top_20_asvs_samples)
```

```{r}
# Get unique values in the specified category
meta_top20 <- META[top_20_asvs_samples$Sample_names]

taxa_top20 <- as.data.frame(taxa[top_20_asvs,])
taxa_top20$ASV_ID <- row.names(taxa_top20)
taxa_top20 <- taxa_top20 %>% arrange(ASV_ID)

write.csv(taxa_top20, "Top20_ASV_CoralBay_SW.csv")
 


#Combine the data frames into one
data_frame_merge <- merge(meta_top20, top_20_asvs_samples,
                          by = 'row.names', all = TRUE)

write.csv(data_frame_merge, "Top20_Short_CB_SW.csv")

#Reshape the data into long format
long_df_CB_SW <- data_frame_merge %>% pivot_longer(
                    cols=starts_with("ASV"),
                    names_to='ASV_ID',
                    values_to='ASV_prop') %>% 
                    arrange(ASV_ID)


mean_per_sample_CB_SW <- long_df_CB_SW %>% 
                  group_by(Sample_ID) %>%
                  summarise(mean = mean(ASV_prop),
                            SD = sd(ASV_prop),
                            sum = sum(ASV_prop))
    # min sum =  0.3800669, max sum = 0.7024470

total_mean_CB_SW <- mean_per_sample_CB_SW %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.5654271, SD = 0.07055103


mean_per_sample_CBandFB <- rbind(mean_per_sample_CB_SW, mean_per_sample_FB_SW)

#mean for coral
total_mean_CB_and_FB <- mean_per_sample_CBandFB %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.5332843, SD = 0.08011193 



palette20 <- c("#C5EC8A",  "#6EB3EB",  "#4CEADD", "#ECF143", "#C571D9", "#962BEF"  , "#DFA643",    "#E64A8A", "#55AE62", "#E45F4E", "#E17FAD", "#61E97D", "#E9D2A9", "#6B6E7F", "#E33EE6", "#ACAA51", "#A5D4E8",  "#BC9DE1",  "#6765F0", "#B99299")





ggplot(long_df_CB_SW, aes(x=Date, y=ASV_prop, fill=ASV_ID)) + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Tag, scale="free") +
  scale_fill_manual(values=palette20) +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  labs(title = "Top 20 ASVs from SW from Coral Bay") +
  xlab(" ")
#%#%#%#%


```
Top20ASVsfromSWfromCoralBay



#Bubble plots
##Seawater from Fish Bay
```{r}
asv_abundance <- as.data.frame(as.matrix(otu_table(ps.seawater.Fishbay)))
 #46 obs of 7023 variables
meta_forBubble <- sample_data(ps.seawater.Fishbay)

ASVofInterst <- c("ASV1", "ASV45", "ASV67", "ASV79")
asv_abun_ofInterest <- asv_abundance[, ASVofInterst]
asv_abun_ofInterest$Sample_names <- row.names(asv_abun_ofInterest)

combo_data <- cbind(asv_abun_ofInterest, meta_forBubble)

colNeeded <- c("ASV1", "ASV45", "ASV67", "ASV79", "Sample_names", "Site", "Tag", "Date", "Sample_Type", "Healthy_Diseased_Recovered")
combo_data <- combo_data[,colNeeded]


#ASV 1
ggplot(combo_data,
       aes(x = Date, 
           y = Tag,
           size = ASV1, 
          color = Healthy_Diseased_Recovered)) +
  geom_point()+
  labs(x = "Date", y = "Colony Tag", title = "ASV 1 Relative Abundance") +
   theme(axis.text.x = element_text(angle = 90, hjust = 1)) 

#ASV 45
ggplot(combo_data,
       aes(x = Date, 
           y = Tag,
           size = ASV45, 
          color = Healthy_Diseased_Recovered)) +
  geom_point()+
  labs(x = "Date", y = "Colony Tag", title = "ASV 45 Relative Abundance") +
   theme(axis.text.x = element_text(angle = 90, hjust = 1)) 

#ASV 67
ggplot(combo_data,
       aes(x = Date, 
           y = Tag,
           size = ASV67, 
          color = Healthy_Diseased_Recovered)) +
  geom_point()+
  labs(x = "Date", y = "Colony Tag", title = "ASV 67 Relative Abundance") +
   theme(axis.text.x = element_text(angle = 90, hjust = 1)) 

#ASV 79
ggplot(combo_data,
       aes(x = Date, 
           y = Tag,
           size = ASV79, 
          color = Healthy_Diseased_Recovered)) +
  geom_point()+
  labs(x = "Date", y = "Colony Tag", title = "ASV 79 Relative Abundance") +
   theme(axis.text.x = element_text(angle = 90, hjust = 1)) 

```



##Seawater from Coral Bay
```{r}
asv_abundance <- as.data.frame(as.matrix(otu_table(ps.Seawater.Coralbay)))
 #45 obs of 9158 variables
meta_forBubble <- sample_data(ps.Seawater.Coralbay)

ASVofInterst <- c("ASV1", "ASV45", "ASV67", "ASV79")
asv_abun_ofInterest <- asv_abundance[, ASVofInterst]
asv_abun_ofInterest$Sample_names <- row.names(asv_abun_ofInterest)

combo_data <- cbind(asv_abun_ofInterest, meta_forBubble)

colNeeded <- c("ASV1", "ASV45", "ASV67", "ASV79", "Sample_names", "Site", "Tag", "Date", "Sample_Type", "Healthy_Diseased_Recovered", "Site_Disease")
combo_data <- combo_data[,colNeeded]

#ASV 1
ggplot(combo_data,
       aes(x = Date, 
           y = Tag,
           size = ASV1, 
          color = Site_Disease)) +
  geom_point()+
  labs(x = "Date", y = "Colony Tag", title = "ASV 1 Relative Abundance") +
   theme(axis.text.x = element_text(angle = 90, hjust = 1)) 

```



#Differentail abundance
This is based on code from Mallory, which she modified from Cynthia's code.
The goal is to conduct differential abundance analyses on the top 200 must abundant ASVs. 
First, I will make a phyloseq object with only healthy and diseased samples, because differential abundance is only done on two factors at a time. Then I will pull out the taxa with the top 200 highest abundance. Then conduct differential abundance on that. 

##Healthy or diseased: Seawater from Fish Bay 
```{r}
#Using a phyloseq object with relative, subset the phyloseq object to just data for healthy or diseased corals (no dead)
ps.SW.Fishbay.HorD <- ps.seawater.Fishbay %>%
  subset_samples(Healthy_or_Diseased != "Dead") %>% #remove dead samples
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

# Make a table of the ASVs in order or most abundance to least abundant. TotAbund shows the total abundance of each ASV across samples. 
SW.Fishbay.HorD.tbl <-
    phyloseq::otu_table(ps.SW.Fishbay.HorD) %>%
    t(.) %>%  # . represents the product of the line above it
    as.data.frame %>%
    dplyr::slice(which(rowSums(.) > 0)) %>%
    as.data.frame %>%
    dplyr::mutate(TotAbund = rowSums(.)) %>%
    dplyr::arrange(desc(TotAbund))

#Get the top 200 most abundace ASVs and get rid of TotAbund
Top200_ASVs <- SW.Fishbay.HorD.tbl[1:200, ] 
Top200_ASVs <- Top200_ASVs  %>%
                dplyr::select(-TotAbund)

Top200_ASVs <- rownames(Top200_ASVs) # Pull out just the ASV names of top 200
```


```{r}
#However, corncob wants count data, not relative abundance. So I am going to use "Top200_ASVs" to filter a new phyloseq object called "ps.Counts.Seawater.Fishbay" which has count data

#Make a Seawater from Fish Bay phyloseq object with count data, not relative abundance
ps.Counts.Seawater.Fishbay <- ps.decontam.filt %>%
  subset_samples(Sample_Type == "Seawater") %>% #subset coral samples
  subset_samples(Site == "Fish_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  subset_samples(Healthy_or_Diseased != "Dead") %>% #remove dead samples
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

#check out the meta data of this phyloseq object to double check that I have the samples I want in it
meta_temp <- sample_data(ps.Counts.Seawater.Fishbay)
rm(meta_temp)
#Double check that this is count data
otu_table(ps.Counts.Seawater.Fishbay)[1:3, 1:3]


#Prune the phyloseq object, so it only has the ASVs in Top200_ASVs. This will give a new phyloseq object with 200 top taxa across samples from healthy or diseased samples
ps.SW.Fishbay_Top200ASVs <- phyloseq::prune_taxa(Top200_ASVs, ps.Counts.Seawater.Fishbay) 
```


```{r}
# Using corncob and Cynthia's code on filtered top 200 taxa data

# Make factors
# ps.SW.Fishbay_Top200ASVs is the phyloseq object with the top 200 taxa across treatments 1 and 3
sample_data(ps.SW.Fishbay_Top200ASVs)$Healthy_or_Diseased <-   factor(sample_data(ps.SW.Fishbay_Top200ASVs)$Healthy_or_Diseased,levels=c("Healthy", "Diseased"))

# Make subsets of the treatments
ps200_HorD <- subset_samples(ps.SW.Fishbay_Top200ASVs, Healthy_or_Diseased %in% c("Healthy", "Diseased"))

set.seed(1)

# Corncob: Differential abundance test comparing T48 samples of treatments 1 and 3 by treatment
ps.SW.Fishbay_Top200ASVs <- differentialTest(formula = ~ Healthy_or_Diseased, 
                             phi.formula = ~ Healthy_or_Diseased,
                             formula_null = ~ 1,
                             phi.formula_null = ~ Healthy_or_Diseased,
                             test = "Wald", boot = FALSE,
                             data = ps200_HorD,
                             fdr_cutoff = 0.05)

ps.SW.Fishbay_Top200ASVs$significant_taxa
length(ps.SW.Fishbay_Top200ASVs$significant_taxa) #10! that's a super reasonable number! Woohoo!
# [1] "ASV45"  "ASV67"  "ASV79"  "ASV84"  "ASV158" "ASV189" "ASV203" "ASV246" "ASV307" "ASV354"
plot(ps.SW.Fishbay_Top200ASVs)
```





##Before or after: Seawater from Coral Bay 
```{r}
#Using a phyloseq object with relative (made in section above)
ps.Seawater.Coralbay

# Make a table of the ASVs in order or most abundance to least abundant. TotAbund shows the total abundance of each ASV across samples. 
SW.CoralBay.tbl <-
    phyloseq::otu_table(ps.Seawater.Coralbay) %>%
    t(.) %>%  # . represents the product of the line above it
    as.data.frame %>%
    dplyr::slice(which(rowSums(.) > 0)) %>%
    as.data.frame %>%
    dplyr::mutate(TotAbund = rowSums(.)) %>%
    dplyr::arrange(desc(TotAbund))

#Get the top 200 most abundace ASVs and get rid of TotAbund
Top200_ASVs <- SW.CoralBay.tbl[1:200, ] 
Top200_ASVs <- Top200_ASVs  %>%
                dplyr::select(-TotAbund)

Top200_ASVs <- rownames(Top200_ASVs) # Pull out just the ASV names of top 200
```


```{r}
#However, corncob wants count data, not relative abundance. So I am going to use "Top200_ASVs" to filter a new phyloseq object called "ps.Counts.Seawater.Fishbay" which has count data

#Make a Seawater from Fish Bay phyloseq object with count data, not relative abundance
ps.Counts.Seawater.CoralBay <- ps.decontam.filt %>%
  subset_samples(Sample_Type == "Seawater") %>% #subset coral samples
  subset_samples(Site == "Coral_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>%
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

#check out the meta data of this phyloseq object to double check that I have the samples I want in it
meta_temp <- sample_data(ps.Counts.Seawater.CoralBay)
rm(meta_temp)
#Double check that this is count data
otu_table(ps.Counts.Seawater.CoralBay)[1:3, 1:3]


#Prune the phyloseq object, so it only has the ASVs in Top200_ASVs. This will give a new phyloseq object with 200 top taxa across samples from healthy or diseased samples
ps.SW.CoralBay_Top200ASVs <- phyloseq::prune_taxa(Top200_ASVs, ps.Counts.Seawater.CoralBay) 
```


```{r}
# Using corncob and Cynthia's code on filtered top 200 taxa data

# Make factors
# ps.SW.Fishbay_Top200ASVs is the phyloseq object with the top 200 taxa across treatments 1 and 3
sample_data(ps.SW.CoralBay_Top200ASVs)$Site_Disease <-   factor(sample_data(ps.SW.CoralBay_Top200ASVs)$Site_Disease,levels=c("Before", "After"))

# Make subsets of the treatments
ps200_BorA <- subset_samples(ps.SW.CoralBay_Top200ASVs, Site_Disease %in% c("Before", "After"))

set.seed(1)

# Corncob: Differential abundance test comparing T48 samples of treatments 1 and 3 by treatment
ps.SW.CoralBay_Top200ASVs <- differentialTest(formula = ~ Site_Disease, 
                             phi.formula = ~ Site_Disease,
                             formula_null = ~ 1,
                             phi.formula_null = ~ Site_Disease,
                             test = "Wald", boot = FALSE,
                             data = ps200_BorA,
                             fdr_cutoff = 0.05)

length(ps.SW.CoralBay_Top200ASVs$significant_taxa) #46! that's kinda big-ish number
ps.SW.CoralBay_Top200ASVs$significant_taxa
      #  [1] "ASV5"   "ASV7"   "ASV9"   "ASV17"  "ASV18"  "ASV24"  "ASV26"  "ASV27"  "ASV34"  "ASV35"  "ASV39"  "ASV40" 
      # [13] "ASV49"  "ASV52"  "ASV62"  "ASV63"  "ASV88"  "ASV89"  "ASV93"  "ASV95"  "ASV101" "ASV107" "ASV125" "ASV129"
      # [25] "ASV137" "ASV141" "ASV148" "ASV149" "ASV157" "ASV162" "ASV164" "ASV165" "ASV185" "ASV212" "ASV225" "ASV242"
      # [37] "ASV252" "ASV261" "ASV262" "ASV265" "ASV277" "ASV368" "ASV436" "ASV450" "ASV579" "ASV827"
plot(ps.SW.CoralBay_Top200ASVs)
```


