---
title: "STJ_SCTLD_Timeseries_Phyloseq_Sept2023"
author: "JeanneBloomberg"
date: "5/27/2023"
output: html_document
---

#Set up
```{r setup, include=FALSE}
setwd("~/Documents/WHOI/Projects/STJ_SCTLD_TS")
```

Reminder for clearing environment environment: you can clear everything except for specific objects with: 
rm(list = setdiff(ls(), c("keep_this_object", "keep_another_object")))


```{r}
# For data wrangling
library(dplyr); packageVersion("dplyr")
library(tidyr); packageVersion("tidyr")

# For microbiome/metagenome analysis
library(corncob)
library(vegan); packageVersion("vegan")
library(phyloseq); packageVersion("phyloseq")
#install.packages("phyloseq", type = "source")
library(emmeans)

# For visualization
library(ggplot2); packageVersion("ggplot2")
library(randomcoloR)

library(cowplot)
library(ggplotify)
library(ggpubr)

library(lme4)
library(lmerTest)
library(readxl)
library(purrr)
```

#Upload data
```{r}
load("~/Documents/WHOI/Projects/STJ_SCTLD_TS/otus_SCTSCTLDTD_J1N2V2_final.RData")

taxa <- as.matrix(read.table("taxonomy_SCTSCTLDTD_J1N2V2_final.txt", sep = "\t", row.names = 1, header = TRUE))

ASV_ID <-read.table("ASVsequences_SCTSCTLDTD_J1N2V2_final.txt", sep = "\t", row.names = 1, header = TRUE)

# metadata <- read.csv("meta_table.csv", header = TRUE)
# #metadata <- metadata[,1:26]
# row.names(metadata) <- metadata$Sample_names

metadata <- read_xlsx("meta_table.xlsx", na="NA")
metadata <- as.data.frame(metadata)
#plot(metadata$Tag ~ metadata$Date)
row.names(metadata) <- metadata$Sample_names
```


#Removes <10,000 reads
Added this processing on Nov. 3, 2023 (orginially in DADA2_STJ_SCTLD_Timeseries_Aug2023.Rmd, but made more sense to me to put it here this time.)
This code is for a special case. In the field, one syringe of seawater was filtered through two different filters, so the the two sequences are actually representative of one sample. I will combine the two samples into one. 
```{r}
#SW 32 and SW 33 are the same water sample, split between two filters
#So I am going to combine the counts from SW-33 to SW-32
head(otus["SW-32",])
  # ASV1 ASV2 ASV3 ASV4 ASV5 ASV6 
  # 882   20   79    0   59   25 
head(otus["SW-33",])
  # ASV1 ASV2 ASV3 ASV4 ASV5 ASV6 
  # 99    0    0    0    0    0 

otus["SW-32",] <- otus["SW-32",] + otus["SW-33",]
head(otus["SW-32",])
    # ASV1 ASV2 ASV3 ASV4 ASV5 ASV6 
    # 981   20   79    0   59   25 
    # OK I think it worked, now I have to remove sample 33 from every phyloseq with Fish Bay Seawater
```


Goal: remove samples with less than 10,000 reads
```{r}

#look at the total reads for each sample
total_reads_per_sample <- data.frame(Sample = row.names(otus),
                                        Reads = rowSums(otus[,]))
keep <- rowSums(otus[,]) >= 10000

removed <- rowSums(otus[,]) < 10000

Samples_removed <- as.data.frame(removed[removed == TRUE])

otus_noLowReads <- otus[keep,]

cat(sum(removed, na.rm = TRUE), 'samples were removed because they had less than 10000 sequences.\n')
  # 25 samples were removed because they had less than 10000 sequences.
  # (but of these, 19 are negative controls)

cat(sum(keep, na.rm = TRUE), 'samples have sufficient reads and thus, remain in the dataset.\n')
  # 194 samples have sufficient reads and thus, remain in the dataset.
```
The non-control samples that are removed are: 
Coral 11 --> losing 
Coral 107 --> losing
SW 33 --> this is a half sample with SW-32, so not losing a data point
SW 34 --> this has a duplicate, so not losing a data point
SW 42 --> this has a duplicate, so not losing a data point
SW 9 --> losing


#Remove duplicates

Some of the samples have duplicates, so I want to pick which duplicate is better to process. 
```{r}
TotalReads <- as.data.frame(rowSums(otus_noLowReads[,]))

#Manually checked which samples has more reads for each remaining duplicate
#Going to remove: 
#  Coral_52_1
#  SW-26
#  SW-35
#  SW-8duplicate

duplicates_toRemove <- c("Coral_52_1", "SW-26", "SW-35", "SW-8duplicate")

otus_noDuplicates <- otus_noLowReads[!(rownames(otus_noLowReads) %in% duplicates_toRemove),]

dim(otus)
  # 219 23901
dim(otus_noLowReads)
  # 194 23901
dim(otus_noDuplicates)
  # 190 23901
```


Ok now I am going to cut down the meta data so that it only includes the samples I am using going forward -- i.e. without samples with low read yields and without duplicates

```{r}
otus <- otus_noDuplicates
samples_we_use <- row.names(otus)
ASV <- as.matrix(otus)

metadata <- metadata[samples_we_use, ]
str(metadata$Date) #check that date is POSIXct
```


#Make base phyloseq object 

```{r}
# Make a phyloseq object. 
ASV = otu_table(ASV, taxa_are_rows = FALSE)
TAX = tax_table(taxa)
META = sample_data(metadata)

ps <- phyloseq(ASV, TAX, META)
  #taxa: 23901
  #samples: 190

#remove contaminants
load("~/Documents/WHOI/Projects/STJ_SCTLD_TS/contam.RData")

ps.decontam <- prune_taxa(!contam$contaminant, ps)#went from 20481 taxa to 20430 taxa
  #taxa: 23857
  #samples: 190

#remove ASVS that are mitochondria or chloroplasts
ps.decontam.nomitochloro <- ps.decontam %>%
    subset_taxa(Family !="Mitochondria" & Order !="Chloroplast")
  #taxa: 22618
  #samples: 190

#remove ASVS with all zeros 
ps.decontam.nozero <- filter_taxa(ps.decontam.nomitochloro, function(x) sum(x) > 0, TRUE)
  #taxa: 22313 
  #samples: 190
    
#Remove the samples I don't want to use: (13 samples removed)
    #Tag 1633 (only one sampling date; 4 additional samples)
    #Tag bonus (only one sampling date; 2 additional samples)
    #Tag 3979 (only one sampling date; 1 additional sample)
    #Tag 2300 (only one sampling date; 2 additional samples)
    #HealthState DH for coral samples (3 additional coral samples; DH ok for seawater)
        # C-105
        # Coral_18
        # C-116
 
 ps.decontam.filt_1 <- ps.decontam.nozero %>%
    subset_samples(Sample_ID !="C-105" & 
                   Sample_ID !="Coral_18" &
                   Sample_ID !="C-116" & 
                   Tag !="1633" & 
                   Tag !="bonus" & 
                   Tag !="3979" & 
                   Tag !="2300")
  #taxa: 22313 taxa
  #samples: 178
   

 Reads_per_sample_1 <- as.data.frame(sample_sums(ps.decontam.filt_1))
 mean(Reads_per_sample_1$`sample_sums(ps.decontam.filt_1)`)  #65397.13
 sd(Reads_per_sample_1$`sample_sums(ps.decontam.filt_1)`)  #64605.68
 sum(Reads_per_sample_1$`sample_sums(ps.decontam.filt_1)`) #11640690
 

```

ok well now I have 14 more samples that have >10,000 reads after removing contaminants and chloro/mito. One of those samples (Coral-13) has 9933 reads, which I'll keep. But I will removed the rest. I am going to use 9000 reads as a cut off in my code, so that I keep Coral-13. 
```{r}
samples_over_9000 <- sample_sums(ps.decontam.filt_1) > 9000
Samples_removed_2ndtime <- as.data.frame(samples_over_9000[samples_over_9000 == FALSE])


ps.decontam.filt <- prune_samples(sample_sums(ps.decontam.filt_1) >= 9000, ps.decontam.filt_1) %>%
                    subset_samples(Sample_Type != "Control") %>%   #take out controls
                    subset_samples(Sample_Type != "Mock")

Reads_per_sample <- as.data.frame(sample_sums(ps.decontam.filt))
mean(Reads_per_sample$`sample_sums(ps.decontam.filt)`)  #67415.79
sd(Reads_per_sample$`sample_sums(ps.decontam.filt)`)  #61745.54
sum(Reads_per_sample$`sample_sums(ps.decontam.filt)`) #10449448
 

write.csv(Reads_per_sample, file = "Reads_per_sample.csv")

MetaData_SamplesWeUse <- data.frame(sample_data(ps.decontam.filt))
write.csv(MetaData_SamplesWeUse, file = "MetaData_SamplesWeUse.csv")

```


#phyloseq obejct with coral and seawater from both sites
rm(list = setdiff(ls(), c("ps.decontam.filt")))
```{r}
ps.SWandCoral.BothSites <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type != "Control") %>%   #take out controls
  subset_samples(Sample_Type != "Mock") %>% #take out mock
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros
```

```{r}
SWandCoral.BothSites.ord <- ordinate(ps.SWandCoral.BothSites, "PCoA", "bray",  trymax = 100, k =3, set.seed(18))

plot_ordination(ps.SWandCoral.BothSites, SWandCoral.BothSites.ord, type="samples", color="Site", shape="Sample_Type") +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(fill = guide_legend(title.position = "top", title.hjust = 0.5),
         shape = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site", shape = "Sample Type") +
  scale_color_manual(values = c("orange3", "royalblue3"), labels = c("Coral Bay", "Fish Bay"))

ALL_metadata <- data.frame(sample_data(ps.SWandCoral.BothSites))
perm <- how(
  blocks = ALL_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.C <- adonis2(distance(ps.SWandCoral.BothSites, method="bray") ~ Sample_Type, 
                         data=ALL_metadata, 
                         perm=perm)
perm.C 
        #              Df SumOfSqs      R2      F Pr(>F)   
        # Sample_Type   1   14.496 0.25898 53.473  0.005 **
        # Residual    153   41.476 0.74102                 
        # Total       154   55.972 1.00000   

#_________________________________________________________________
ALL_metadata <- data.frame(sample_data(ps.SWandCoral.BothSites))
perm <- how(
  blocks = ALL_metadata$Date
  )
perm.C <- adonis2(distance(ps.SWandCoral.BothSites, method="bray") ~ Site, 
                         data=ALL_metadata, 
                         perm=perm)
perm.C 
#tag is fignificant


```


#phyloseq obejct with Coral from both sites
rm(list = setdiff(ls(), c("ps.decontam.filt")))
```{r}
ps.BothSites.Coral <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Coral") %>%   #subset coral samples
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

Coral.BothSites.ord <- ordinate(ps.BothSites.Coral, "PCoA", "bray",  trymax = 100, k =3, set.seed(18))
```

From ChatGPT: 
You can extract the x and y coordinates of the PCoA produced by ordinate() using the scores() function. Since your ordinate() call uses "PCoA", the coordinates are stored in Coral.BothSites.ord$vectors.
```{r}
#I want to look at the axes store in Coral.BothSites.ord, which should include more than just the two plotted
#head(Coral.BothSites.ord$vectors) #ok there are 63 axes

#extract the top 3 axes for graphical inspection, which I will use to determine which axes to use in my LMM
PCoA_df <- data.frame(SampleID = rownames(Coral.BothSites.ord$vectors), 
                      Axis1 = Coral.BothSites.ord$vectors[, 1], 
                      Axis2 = Coral.BothSites.ord$vectors[, 2], 
                      Axis3 = Coral.BothSites.ord$vectors[, 3], 
                      Site = sample_data(ps.BothSites.Coral)$Site, 
                      Date = sample_data(ps.BothSites.Coral)$Date, 
                      Tag = sample_data(ps.BothSites.Coral)$Tag)
str(PCoA_df$Date) #confirm that date is still in POSIXct
PCoA_df$Date_scaled <- scale(PCoA_df$Date) #scaled dates around 0
PCoA_df$Days_Since_First <- as.numeric(as.Date(PCoA_df$Date) - as.Date("2020-07-16")) #number of days since the first sampling date
str(PCoA_df$Days_Since_First) #num

```

Graphical inspection --> chose Axis 2 for LM
```{r}
#compare axes 1 and 2 --> there is some clustering along Axis2
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=Site)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(values = c("orange3", "royalblue3"), labels = c("Coral Bay", "Fish Bay"))

#compare axes 1 and 3 --> there is some clustering along Axis3
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Site)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(values = c("orange3", "royalblue3"), labels = c("Coral Bay", "Fish Bay"))

#compare axes 2 and 3 --> more clustering around Axis2, so I am going to use that one for downstream analyses
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=Site)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(values = c("orange3", "royalblue3"), labels = c("Coral Bay", "Fish Bay"))
```

LM with Axis 2 response (coral tissue, testing effect of site)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)
#from MAPB: linear regression -- with time and coral ID and site as predictors. Looking at site but adjust for time and tag and the interaction between tag and time.
#date is in the model as days after the first sampling date of the whole experiment (2020-07-16)
mod <- lm(Axis2 ~ Site + Days_Since_First + Tag + Tag*Days_Since_First, data = PCoA_df) 
summary(mod)
      #                            Estimate Std. Error t value Pr(>|t|)  
      # (Intercept)              -1.510e-01  9.509e-02  -1.588   0.1201  
      # SiteFish_Bay        x      3.000e-01  1.161e-01   2.585   0.0135 *

      # Multiple R-squared:  0.4007,	Adjusted R-squared:  0.05604 
      # F-statistic: 1.163 on 23 and 40 DF,  p-value: 0.33

#write.csv(PCoA_df, "Jeanne_Coral_Data.csv")

```
Result: Microbial communities of coral tissue at Fish Bay and Coral Bay are significantly different (LM: β = 0.300, SE = 0.116, t = 2.585, p = 0.0135)


#phyloseq obejct with SW from both sites
```{r}
rm(list = setdiff(ls(), c("ps.decontam.filt")))

ps.seawater <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Seawater") %>%   #subset samples
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

SW.BothSites.ord <- ordinate(ps.seawater, "PCoA", "bray",  trymax = 100, k =3, set.seed(18))
```

Extract the x and y coordinates of the PCoA produced by ordinate() 
```{r}
#extract the top 3 axes for graphical inspection, which I will use to determine which axes to use in my LMM
PCoA_df <- data.frame(SampleID = rownames(SW.BothSites.ord$vectors), 
                      Axis1 = SW.BothSites.ord$vectors[, 1], 
                      Axis2 = SW.BothSites.ord$vectors[, 2], 
                      Axis3 = SW.BothSites.ord$vectors[, 3], 
                      Site = sample_data(ps.seawater)$Site, 
                      Date = sample_data(ps.seawater)$Date, 
                      Tag = sample_data(ps.seawater)$Tag)
str(PCoA_df$Date) #confirm that date is still in POSIXct
PCoA_df$Date_scaled <- scale(PCoA_df$Date) #scaled dates around 0
PCoA_df$Days_Since_First <- as.numeric(as.Date(PCoA_df$Date) - as.Date("2020-07-16")) #number of days since the first sampling 
str(PCoA_df$Days_Since_First) #num
```

Graphical inspection --> chose Axis 2 for LM
```{r}
#compare axes 1 and 2 --> lots of clustering along Axis2
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=Site)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(values = c("orange3", "royalblue3"), labels = c("Coral Bay", "Fish Bay"))

#compare axes 1 and 3 --> not as much clustering
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Site)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(values = c("orange3", "royalblue3"), labels = c("Coral Bay", "Fish Bay"))

#compare axes 2 and 3 --> more clustering around Axis2, so I am going to use that one for downstream analyses
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=Site)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(values = c("orange3", "royalblue3"), labels = c("Coral Bay", "Fish Bay"))
```

LM with Axis 2 response (all seawater, testing effect of site)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

mod <- lm(Axis2 ~ Site + Days_Since_First + Tag + Tag*Days_Since_First, data = PCoA_df) 
summary(mod)
    #                            Estimate Std. Error t value Pr(>|t|)   
    # (Intercept)               8.348e-02  5.265e-02   1.586  0.11755   
    # SiteFish_Bay             -2.311e-01  6.799e-02  -3.399  0.00114 **

    # Multiple R-squared:  0.7209,	Adjusted R-squared:  0.6251 
    # F-statistic: 7.523 on 23 and 67 DF,  p-value: 4.379e-11

```
Result: Microbial communities of near-coral seawater at Fish Bay and Coral Bay are significantly different (LM: β = -0.231, SE = 0.068, t = -3.399, p = 0.00114). 



#phylo Coral, Fish Bay
rm(list = setdiff(ls(), c("ps.decontam.filt")))
```{r}
ps.coral.Fishbay <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Coral") %>% #subset coral samples
  subset_samples(Site == "Fish_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

Coral.FishBay.ord <- ordinate(ps.coral.Fishbay, "PCoA", "bray",  trymax = 100, k =3, set.seed(18))
```

Figure 2A for plotting. Fish Bay, coral tissue x HDR. 
```{r}
plot2A <- plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", color="Healthy_Diseased_Recovered") +
  geom_point(size = 4, aes(color = factor(Healthy_Diseased_Recovered))) +
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm"),
        plot.title = element_text(hjust = 0.5)) +
  labs(title  = "Coral health state") +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Coral health state", tag = "A") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3"))
```

Extract the x and y coordinates of the PCoA produced by ordinate(). Fish Bay, coral tissue x HDR. 
```{r}
#extract the top 3 axes for graphical inspection, which I will use to determine which axes to use in my LMM
PCoA_df <- data.frame(SampleID = rownames(Coral.FishBay.ord$vectors), 
                      Axis1 = Coral.FishBay.ord$vectors[, 1], 
                      Axis2 = Coral.FishBay.ord$vectors[, 2], 
                      Axis3 = Coral.FishBay.ord$vectors[, 3], 
                      Site = sample_data(ps.coral.Fishbay)$Site, 
                      Date = sample_data(ps.coral.Fishbay)$Date, 
                      Tag = sample_data(ps.coral.Fishbay)$Tag, 
                      HDR = sample_data(ps.coral.Fishbay)$Healthy_Diseased_Recovered)
str(PCoA_df$Date) #confirm that date is still in POSIXct
PCoA_df$Date_scaled <- scale(PCoA_df$Date) #scaled dates around 0
PCoA_df$Days_Since_First <- as.numeric(as.Date(PCoA_df$Date) - as.Date("2020-07-16")) #number of days since the first sampling 
str(PCoA_df$Days_Since_First) #num
```

Graphical inspection. Fish Bay, coral tissue x HDR. chose Axis 1 for LM
```{r}
#compare axes 1 and 2 --> there is some clustering along Axis1, but not a ton 
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=HDR)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3"))

#compare axes 1 and 3 --> not much clustering
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=HDR)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3"))

#compare axes 2 and 3 --> not much clustering
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=HDR)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3"))
```

LM with Axis 1 response (Fish Bay, coral tissue, testing effect of HDR)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

PCoA_df$HDR <- as.factor(PCoA_df$HDR)
PCoA_df$HDR <- relevel(PCoA_df$HDR, ref = "Healthy")

mod <- lm(Axis1 ~ HDR + Days_Since_First + Tag + Tag*Days_Since_First, data = PCoA_df) 
summary(mod)
      #                            Estimate Std. Error t value Pr(>|t|)   
      # (Intercept)              -0.5778545  0.2579787  -2.240  0.03423 * 
      # HDRDiseased              -0.1288451  0.0855321  -1.506  0.14450   
      # HDRRecovered             -0.2912746  0.0930288  -3.131  0.00440 **

      # Multiple R-squared:  0.5527,	Adjusted R-squared:  0.3201 
      # F-statistic: 2.376 on 13 and 25 DF,  p-value: 0.03056

```
HDR has an effect on microbial communities of coral tissue (F(sub 13, 25) = 2.376, p = 0.03056).

-------------------------

Figure 2B for plotting. Fish Bay, coral tissue x Date 
Figure 3 for plotting.  Fish Bay, coral tissue x Date, faceted by Tag
```{r}
sample_data(ps.coral.Fishbay)$Date <- as.Date(sample_data(ps.coral.Fishbay)$Date)
plot2B <- plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", color="Date") +
  geom_point(size = 4) +
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm"),
        plot.title = element_text(hjust = 0.5)) +
  labs(title = "Sampling date") +
  guides(color = guide_legend(title.position = "top",
                              title.hjust = 0.5,
                              ncol = 2)) +
  labs(color = "Date", tag ="B")


#get just the legend for plotting
legend_2Date <- get_plot_component(plot2B, 'guide-box-bottom', return_all = TRUE)
legend_2Date <- as_ggplot(legend_2Date)

     


plot3 <- plot_ordination(ps.coral.Fishbay, Coral.FishBay.ord , type="samples", shape ="Healthy_Diseased_Recovered", color = "Date") +
  facet_wrap(~Tag, scale="fixed", ncol=3)  +
  geom_point(size = 5) +
  theme_bw() +
  guides(color = guide_legend(title.position = "top",
                              title.hjust = 0.5,
                              ncol = 2)) +
  theme(text=element_text(size=16),  
        legend.text=element_text(size=16), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 14), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        panel.spacing = unit(0.75, "cm")) +
  labs(color = "Date", shape = "Coral health state") + 
  scale_shape_discrete(breaks=c("Healthy", "Diseased", "Recovered"))


```

Graphical inspection. Fish Bay, coral tissue x Date. chose Axis 1 for LM
```{r}
#compare axes 1 and 2 --> a litte bit of clustering along Axis 1 and 2
PCoA_df$Days_Since_First <- as.factor(PCoA_df$Days_Since_First) #only do this for visualization purposes. Need to change it back to num before LM.
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 1 and 3 --> stronger clustering along Axis 1
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 2 and 3 --> structure a little stronger in Axis 2, but not as strong as Axis 1 in 1v3 --> so choosing Axis 1
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))
```

LM with Axis 1 response (Fish Bay, coral tissue, testing effect of Date)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

mod <- lm(Axis1 ~ Days_Since_First + Tag, data = PCoA_df) 
      #                    Estimate Std. Error t value Pr(>|t|)  
      # (Intercept)      -1.239e-01  1.057e-01  -1.172   0.2497  
      # Days_Since_First -7.472e-05  6.875e-05  -1.087   0.2852  
```
Date does not have an effect on microbial communities of coral tissue (F(sub 11, 27) = 1.448, p = 0.2091).



#phylo Seawater, Fish Bay
rm(list = setdiff(ls(), c("ps.decontam.filt")))
```{r}
ps.seawater.Fishbay <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Seawater") %>% #subset coral samples
  subset_samples(Site == "Fish_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

SW.FishBay.ord <- ordinate(ps.seawater.Fishbay, "PCoA", "bray",  trymax = 200, weakties = FALSE, set.seed(18))
```

Figure 2C for plotting. Fish Bay, seawater x HDR. 
```{r}
plot2C <- plot_ordination(ps.seawater.Fishbay, SW.FishBay.ord , type="samples", color="Healthy_Diseased_Recovered") +
  geom_point(size = 4, aes(color = factor(Healthy_Diseased_Recovered))) +
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Coral health state", tag = "C") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered", "Dead"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "gray50")) 



#get just the legend for plotting
legend_2health <- get_plot_component(plot2C, 'guide-box-bottom', return_all = TRUE)
legend_2health <- as_ggplot(legend_2health)
```

Extract the x and y coordinates of the PCoA produced by ordinate(). Fish Bay, seawater x HDR. 
```{r}
#extract the top 3 axes for graphical inspection, which I will use to determine which axes to use in my LM
PCoA_df <- data.frame(SampleID = rownames(SW.FishBay.ord$vectors), 
                      Axis1 = SW.FishBay.ord$vectors[, 1], 
                      Axis2 = SW.FishBay.ord$vectors[, 2], 
                      Axis3 = SW.FishBay.ord$vectors[, 3], 
                      Axis4 = SW.FishBay.ord$vectors[, 4],
                      Site = sample_data(ps.seawater.Fishbay)$Site, 
                      Date = sample_data(ps.seawater.Fishbay)$Date, 
                      Tag = sample_data(ps.seawater.Fishbay)$Tag, 
                      HDR = sample_data(ps.seawater.Fishbay)$Healthy_Diseased_Recovered)
str(PCoA_df$Date) #confirm that date is still in POSIXct
PCoA_df$Date_scaled <- scale(PCoA_df$Date) #scaled dates around 0
PCoA_df$Days_Since_First <- as.numeric(as.Date(PCoA_df$Date) - as.Date("2020-07-16")) #number of days since the first sampling 
str(PCoA_df$Days_Since_First) #num
```

Graphical inspection. Fish Bay, seawater x HDR chose Axis 2 for LM
```{r}
#compare axes 1 and 2 --> there is clustering along Axis2, but Axis 1 has outliers
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=HDR)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "grey"))

#compare axes 1 and 3 --> not much clustering along Axis3, but there are outliers driving clustering in Axis 1
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=HDR)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "grey"))

#compare axes 2 and 3 --> more cluster along Axis 2
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=HDR)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Site") +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "grey"))
```

LM with Axis 1 response (Fish Bay, seawater, testing effect of HDR)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

PCoA_df$HDR <- as.factor(PCoA_df$HDR)
PCoA_df$HDR <- relevel(PCoA_df$HDR, ref = "Healthy")

mod <- lm(Axis1 ~ HDR + Days_Since_First + Tag + Tag*Days_Since_First, data = PCoA_df) 
summary(mod)
    #                            Estimate Std. Error t value Pr(>|t|)    
    # (Intercept)               0.4807126  0.3101211   1.550   0.1313    
    # HDRDead                   0.2580687  0.2475132   1.043   0.3052    
    # HDRDiseased              -0.4196322  0.0887766  -4.727 4.69e-05 ***
    # HDRRecovered             -0.2499445  0.1073523  -2.328   0.0266 * 

    # Multiple R-squared:  0.5956,	Adjusted R-squared:  0.413 
    # F-statistic: 3.261 on 14 and 31 DF,  p-value: 0.002981
```
HDR has an effect on microbial communities of neat-coral seawater (F(sub 14, 31) = 3.261, p = 0.002981).

-------------------------

Figure 2D for plotting. Fish Bay, seawater x Date 
```{r}
plot2D <- plot_ordination(ps.seawater.Fishbay, SW.FishBay.ord , type="samples", color="Date") +
  geom_point(size = 4) +
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Date", tag ="D")

```

Graphical inspection. Fish Bay, seawater x Date. chose axis 2
```{r}
#compare axes 1 and 2 --> strong cluserting along axis 2
PCoA_df$Days_Since_First <- as.factor(PCoA_df$Days_Since_First) #only do this for visualization purposes. Need to change it back to num before LM.
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 1 and 3 --> strong clustering along Axis 3
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 2 and 3 --> stronger clustering along Axis 2
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))
```

LM with Axis 2 response (Fish Bay, seawater, testing effect of Date)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

mod <- lm(Axis2 ~ Days_Since_First + Tag, data = PCoA_df) 
      #                    Estimate Std. Error t value Pr(>|t|)   
      # (Intercept)      -1.681e-01  8.787e-02  -1.913  0.06310 . 
      # Days_Since_First  1.561e-04  5.331e-05   2.928  0.00568 **
```
Date has an effect on microbial communities of neat-coral seawater β = 0.2241, SE = 0.1037, t = 2.162, p = 0.0378)

-------------------------
I want to test if tag has an effect for relative abundance stuff

Graphical inspection. Fish Bay, seawater x Tag chose axis 1
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag) 

#compare axes 1 and 2 --> outliers
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=Tag)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 1 and 3 --> outliers
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Tag)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 2 and 3 --> not much clustering
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=Tag)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 2 and 4 --> not much clustering
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis4, color=Tag)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 3 and 4 --> not much clustering
ggplot(data = PCoA_df, aes(x=Axis3, y=Axis4, color=Tag)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 


#compare axes 1 and 4 --> outliers
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis4, color=Tag)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

```

LM with Axis 1 response (Fish Bay, seawater, testing effect of Tag)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

mod <- lm(Axis2 ~ Tag + Days_Since_First + Tag*Days_Since_First, data = PCoA_df) 
summary(mod)
```
Tag has an effect of microbial communities of near-coral seawater.







#phylo Seawater, Coral Bay
rm(list = setdiff(ls(), c("ps.decontam.filt")))
```{r}
ps.Seawater.Coralbay <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Seawater") %>% #subset Seawater samples
  subset_samples(Site == "Coral_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

SW.CoralBay.ord <- ordinate(ps.Seawater.Coralbay, "PCoA", "bray",  trymax = 50, k =2, set.seed(18))
```

Figure 1C for plotting. Coral Bay, seawater x Site_disease 
```{r}
plot1C <- plot_ordination(ps.Seawater.Coralbay, SW.CoralBay.ord , type="samples", color="Site_Disease") +
  geom_point(size = 4, aes(color = factor(Site_Disease))) +
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "C") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"), 
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 
    
#get just the legend for plotting
legend_ReefHealthState <- get_plot_component(plot1C, 'guide-box-bottom', return_all = TRUE)
legend_ReefHealthState <- as_ggplot(legend_ReefHealthState)
```

Extract the x and y coordinates of the PCoA produced by ordinate(). Coral Bay, seawater x Site_disease 
```{r}
#extract the top 3 axes for graphical inspection, which I will use to determine which axes to use in my LM
PCoA_df <- data.frame(SampleID = rownames(SW.CoralBay.ord$vectors), 
                      Axis1 = SW.CoralBay.ord$vectors[, 1], 
                      Axis2 = SW.CoralBay.ord$vectors[, 2], 
                      Axis3 = SW.CoralBay.ord$vectors[, 3],
                      Axis4 = SW.CoralBay.ord$vectors[, 4], 
                      Axis5 = SW.CoralBay.ord$vectors[, 5], 
                      Axis6 = SW.CoralBay.ord$vectors[, 6],
                      Axis7 = SW.CoralBay.ord$vectors[, 7],
                      Axis8 = SW.CoralBay.ord$vectors[, 8],
                      Axis9 = SW.CoralBay.ord$vectors[, 9],
                      Axis10 = SW.CoralBay.ord$vectors[, 10],
                      Site = sample_data(ps.Seawater.Coralbay)$Site, 
                      Date = sample_data(ps.Seawater.Coralbay)$Date, 
                      Tag = sample_data(ps.Seawater.Coralbay)$Tag, 
                      HDR = sample_data(ps.Seawater.Coralbay)$Healthy_Diseased_Recovered, 
                      Site_Disease  = sample_data(ps.Seawater.Coralbay)$Site_Disease)
str(PCoA_df$Date) #confirm that date is still in POSIXct
PCoA_df$Date_scaled <- scale(PCoA_df$Date) #scaled dates around 0
PCoA_df$Days_Since_First <- as.numeric(as.Date(PCoA_df$Date) - as.Date("2020-07-16")) #number of days since the first sampling 
str(PCoA_df$Days_Since_First) #num
```

Graphical inspection. Coral Bay, seawater x Site_disease chose Axis 2 for LM
```{r}
#compare axes 1 and 2 --> clustering on both, about the same
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=Site_Disease)) +
  geom_point(size = 4) +  
  theme_bw() +
   theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "C") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"), 
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 

#compare axes 1 and 3 --> clustering on both, a little more on Axis 1
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Site_Disease)) +
  geom_point(size = 4) +  
  theme_bw() +
   theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "C") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"), 
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 

#compare axes 2 and 3 --> more cluster along Axis 2
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=Site_Disease)) +
  geom_point(size = 4) +  
  theme_bw() +
   theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "C") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"), 
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 
```

LM with Axis 2 response (Coral Bay, seawater, testing effect of Site_disease)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

mod <- lm(Axis2 ~ Site_Disease + Days_Since_First + Tag + Tag*Days_Since_First, data = PCoA_df) 
summary(mod)
      #                            Estimate Std. Error t value Pr(>|t|)    
      # (Intercept)              -2.157e-01  7.567e-02  -2.851  0.00757 ** 
      # Site_DiseaseBefore        2.346e-01  4.624e-02   5.074  1.6e-05 ***

      # Multiple R-squared:  0.5159,	Adjusted R-squared:  0.3343 
      # F-statistic: 2.842 on 12 and 32 DF,  p-value: 0.009092
```
Site_disease has an effect on the near-coral seawater at Coral Bay (F(sub 12, 32) = 2.842, p = 0.009092)

-------------------------

Figure 1D for plotting. Coral Bay, seawater x Date 
```{r}
plot1D <- plot_ordination(ps.Seawater.Coralbay, SW.CoralBay.ord , type="samples", color="Date") +
  geom_point(size = 4) +
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0, 0, 0.5), "cm")) +
  guides(color = guide_legend(title.position = "top", 
                              title.hjust = 0.5,
                              ncol = 2)) +
  labs(color = "Date", tag ="D")


#get just the legend for plotting
legend_Date1 <- get_plot_component(plot1D, 'guide-box-bottom', return_all = TRUE)
legend_Date1 <- as_ggplot(legend_Date1)


perm <- how(
  blocks = SW_CoralBay_metadata$Tag, # Restrict permutations within each block (Tag)
  within = Within(type = "series")  # Keep the time series structure within each plot
)
perm.SW.CB.date <- adonis2(distance(ps.Seawater.Coralbay, method="bray") ~ Date, 
                           data=SW_CoralBay_metadata, 
                          perm=perm)
perm.SW.CB.date 

#          Df SumOfSqs      R2      F Pr(>F)   
# Date     11   2.8474 0.85168 17.227  0.005 **
# Residual 33   0.4959 0.14832                 
# Total    44   3.3432 1.00000  

```

Graphical inspection. Corak Bay, seawater x Date. Chose Axis 3
```{r}
PCoA_df$Days_Since_First <- as.factor(PCoA_df$Days_Since_First) #only do this for visualization purposes. Need to change it back to num before LM.

#I did a lot of interations, so I wrote down results 
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "none", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 
```

LM with Axis 3 response (Coral Bay, seawater, testing effect of Date)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

mod <- lm(Axis3 ~ Days_Since_First + Tag , data = PCoA_df) 
summary(mod)
      #                  Estimate Std. Error t value Pr(>|t|)   
      # (Intercept)       0.060910   0.039939   1.525  0.13552   
      # Days_Since_First -0.015237   0.004657  -3.272  0.00228 **
```
Date has an effect on the near-coral seawater at Coral Bay 














#phylo Coral, Coral Bay
rm(list = setdiff(ls(), c("ps.decontam.filt")))
```{r}
ps.Coral.Coralbay <- ps.decontam.filt %>%
  transform_sample_counts(function(x) x / sum(x) ) %>%  #this makes everything in relative abundances
  subset_samples(Sample_Type == "Coral") %>% #subset Seawater samples
  subset_samples(Site == "Coral_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros

Coral.CoralBay.ord <- ordinate(ps.Coral.Coralbay, "PCoA", "bray",  trymax = 50, k =2, set.seed(18))
```

Figure 1A for plotting. Coral Bay,  coral tissue x Site_Disease 
```{r}
plot1A <- plot_ordination(ps.Coral.Coralbay, Coral.CoralBay.ord , type="samples", color="Site_Disease") +
  geom_point(size = 4, aes(color = factor(Site_Disease))) +
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "A", title  = "Coral health state") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"),
                       labels = c("Before SCTLD arrived", "After SCTLD arrived"))
```

Extract the x and y coordinates of the PCoA produced by ordinate(). Coral Bay, coral tissue x Site_disease 
```{r}
#extract the top 3 axes for graphical inspection, which I will use to determine which axes to use in my LM
PCoA_df <- data.frame(SampleID = rownames(Coral.CoralBay.ord$vectors), 
                      Axis1 = Coral.CoralBay.ord$vectors[, 1], 
                      Axis2 = Coral.CoralBay.ord$vectors[, 2], 
                      Axis3 = Coral.CoralBay.ord$vectors[, 3], 
                      Site = sample_data(ps.Coral.Coralbay)$Site, 
                      Date = sample_data(ps.Coral.Coralbay)$Date, 
                      Tag = sample_data(ps.Coral.Coralbay)$Tag, 
                      HDR = sample_data(ps.Coral.Coralbay)$Healthy_Diseased_Recovered, 
                      Site_Disease  = sample_data(ps.Coral.Coralbay)$Site_Disease)
str(PCoA_df$Date) #confirm that date is still in POSIXct
PCoA_df$Date_scaled <- scale(PCoA_df$Date) #scaled dates around 0
PCoA_df$Days_Since_First <- as.numeric(as.Date(PCoA_df$Date) - as.Date("2020-07-16")) #number of days since the first sampling 
str(PCoA_df$Days_Since_First) #num
```

Graphical inspection. Coral Bay, coral tissue x Site_disease chose Axis 1 for LM
```{r}
#compare axes 1 and 2 --> clustering on axis 1
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=Site_Disease)) +
  geom_point(size = 4) +  
  theme_bw() +
   theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "C") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"), 
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 

#compare axes 1 and 3 --> clustering on axis 1
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Site_Disease)) +
  geom_point(size = 4) +  
  theme_bw() +
   theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "C") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"), 
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 

#compare axes 2 and 3 --> not much clustering. 
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=Site_Disease)) +
  geom_point(size = 4) +  
  theme_bw() +
   theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", legend.direction ="vertical", 
        plot.margin = unit(c(0.5, 0.5, 0, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) +
  labs(color = "Reef health state", tag = "C") +
  scale_color_manual(breaks=c("Before", "After"), 
                       values = c("dodgerblue3", "hotpink2"), 
                       labels = c("Before SCTLD arrived", "After SCTLD arrived")) 
```

LM with Axis 1 response (Coral Bay, coral tissue, testing effect of Site_disease)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

mod <- lm(Axis1 ~ Site_Disease + Days_Since_First + Tag + Tag*Days_Since_First, data = PCoA_df) 
summary(mod)
    #                            Estimate Std. Error t value Pr(>|t|)
    # (Intercept)              -0.1053671  0.2439276  -0.432    0.673
    # Site_DiseaseBefore        0.2637826  0.1659972   1.589    0.138

    # Multiple R-squared:  0.4531,	Adjusted R-squared:  -0.09384 
    # F-statistic: 0.8284 on 12 and 12 DF,  p-value: 0.6252
```
Site_disease does not have an effect on coral tissue communities (F(sub 12, 23) = 0.8284, p = 0.6252)


-------------------------

Figure 1b for plotting. Coral Bay, tissue x Date 
```{r}
plot1B <- plot_ordination(ps.Coral.Coralbay, Coral.CoralBay.ord , type="samples", color="Date") +
  geom_point(size = 4) +
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0, 0.5, 0.5), "cm")) +
  guides(color = guide_legend(title.position = "top", 
                              title.hjust = 0.5,
                              ncol = 2)) +
  labs(color = "Date", tag = "B", title  = "Sampling date")

```

Graphical inspection. Corak Bay, tissue x Date. Chose Axis 1.
```{r}
#compare axes 1 and 2 --> clustering on Axis 1
PCoA_df$Days_Since_First <- as.factor(PCoA_df$Days_Since_First) #only do this for visualization purposes. Need to change it back to num before LM.
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis2, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 1 and 3 --> clustering on both, more for Axis 3
ggplot(data = PCoA_df, aes(x=Axis1, y=Axis3, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5)) 

#compare axes 2 and 3 --> more on Axis 3
ggplot(data = PCoA_df, aes(x=Axis2, y=Axis3, color=Days_Since_First)) +
  geom_point(size = 4) +  
  theme_bw() +
  theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm")) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))
```

LM with Axis 1 response (Coral Bay, coral tissue, testing effect of Date)
```{r}
PCoA_df$Tag <- as.factor(PCoA_df$Tag)
str(PCoA_df$Tag)
PCoA_df$Days_Since_First <- as.numeric(PCoA_df$Days_Since_First)
str(PCoA_df$Days_Since_First)

mod <- lm(Axis1 ~  Days_Since_First + Tag, data = PCoA_df) 
summary(mod)
      #                    Estimate Std. Error t value Pr(>|t|)
      # (Intercept)      -0.0179973  0.1067308  -0.169    0.868
      # Days_Since_First -0.0000974  0.0001702  -0.572    0.574
```
Date does not have an effect on coral tissue communities ((F(sub 11, 13) = 0.6034, p = 0.796))




##put plots together for PCoA @ CB
```{r}

plot_grid(plot1A + theme(legend.position = "none"), 
          plot1B + theme(legend.position = "none"), 
          plot1C + theme(legend.position = "none"), 
          plot1D + theme(legend.position = "none"), 
          legend_ReefHealthState, 
          legend_Date1, 
          ncol = 2, nrow = 3)

```

##put plots together for PCoA @ FB
```{r}

plot_grid(plot2A + theme(legend.position = "none"), 
          plot2B + theme(legend.position = "none"), 
          plot2C + theme(legend.position = "none"), 
          plot2D + theme(legend.position = "none"), 
          legend_2health, 
          legend_2Date, 
          ncol = 2, nrow = 3)

```









#SCTLD Prevalence
```{r}
Prevelance_ForAnalyses <- read.csv("~/Documents/WHOI/Projects/STJ_SCTLD_TimeseriesExperiment/Disease Wheel Data/Prevelance_ForAnalyses.csv")
Prevelance_ForAnalyses$Date_nice <- as.Date(Prevelance_ForAnalyses$Date) #make ggplot know it's a date

unique(Prevelance_ForAnalyses$Date_nice)

ggplot()+
  geom_point(data=Prevelance_ForAnalyses, aes(x = Date_nice, y=Prev_Percent), na.rm = TRUE, color="Red") +
  geom_smooth(data=Prevelance_ForAnalyses, aes(x = Date_nice, y=Prev_Percent), na.rm = TRUE, method = loess, color="Red", se = FALSE) +
  scale_y_continuous(name = "SCTLD Prevelance (percent)") +
  xlab("Date") +
  theme(text=element_text(size=13)) +
  ggtitle("SCTLD Prevelance combined") 

#For Loess:
#default span is α = 0.75


prev_fishbay <- subset(Prevelance_ForAnalyses, Site == "Fish_Bay")
plot_FB <- ggplot()+
  geom_point(data=prev_fishbay, aes(x = Date_nice, y=Prev_Percent), na.rm = TRUE, color="red3", size = 4) +
  geom_smooth(data=prev_fishbay, aes(x = Date_nice, y=Prev_Percent), na.rm = TRUE, method = loess, color="red3", se = FALSE, size=2) +
  scale_y_continuous(name = "SCTLD Prevelance (percent)") +
  xlab("") +
  labs(tag = "B") +
  theme_classic() +
  theme(text=element_text(size=16))+
  ggtitle("Fish Bay") 


prev_coralbay <- subset(Prevelance_ForAnalyses, Site == "Coral_Bay")
plot_CB <- ggplot()+
  geom_point(data=prev_coralbay, aes(x = Date_nice, y=Prev_Percent), na.rm = TRUE, color="red3", size = 4) +
  geom_smooth(data=prev_coralbay, aes(x = Date_nice, y=Prev_Percent), na.rm = TRUE, method = loess, color="red3", se = FALSE, size=2) +
  scale_y_continuous(name = "SCTLD Prevelance (percent)") +
  xlab("") +
  labs(tag = "A") +
  theme_classic() +
  theme(text=element_text(size=16))+
  ggtitle("Coral Bay") 


plot_grid(plot_CB, 
          plot_FB, 
          ncol = 2, nrow = 1, 
          align = 'h', 
          rel_widths = c(1, 1))

```














#Beta Diversity 
useful resource on beta div types: https://docs.onecodex.com/en/articles/4150649-beta-diversity

## Means: SW from Fish Bay
I am trying to find the mean beta diversity for either each time point, or for the epidemic vs endemic phases
```{r}
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.seawater.Fishbay", "ps.Seawater.Coralbay", "Prevelance_ForAnalyses", "prev_coralbay", "prev_fishbay")))
```

Start out with seawater from Fish Bay
```{r}
#using: ps.seawater.Fishbay

otu_rel <- as.data.frame(otu_table(ps.seawater.Fishbay)) #pull out OTU table
  otu_rel[1:3,1:3] #make sure sample names are rows
metadat_rel <- as.data.frame(sample_data(ps.seawater.Fishbay)) #pull out Meta table

#Check if my rows are in the same order
identical(rownames(otu_rel), rownames(metadat_rel))
    #True!

#make a distance matrix, and specify the method - in this case, bray curtis 
# see https://rdrr.io/cran/vegan/man/vegdist.html
# how dissimilar is each sample is from every other sample
Veg <- vegdist(otu_rel, method="bray") 

# model that that we use to test the differences in distance to the centroid 
# takes the matrix and figues out how dispersed the samples are within groups
# Quantifies the dispersion between each of the points 
# How dispersed are the groups -- Find the centroid then report how far each point is from the centroid
Mod_BetaDisp <-betadisper(Veg, metadat_rel$Healthy_Diseased_Recovered)

Mod_BetaDisp
      # Average distance to median:
      #    Dead  Diseased   Healthy Recovered 
      #  0.2476    0.4665    0.2408    0.2468 
summary(Mod_BetaDisp)

Mod_aov <- anova(Mod_BetaDisp)
Mod_aov
    # Response: Distances
    #           Df  Sum Sq  Mean Sq F value    Pr(>F)    
    # Groups     3 0.46078 0.153595  7.4748 0.0004052 ***
    # Residuals 42 0.86304 0.020548
    #this shows that HDR has an significant effect on beta div. 

#ok but these models do not account for the nested/nonindependent nature of the data, but it is useful for data exploration purposes only.

```

Prepare for plotting
```{r}
dist.tab_SW_FB <- as.data.frame(Mod_BetaDisp$distances)
dist.tab_SW_FB$Sample_names <- rownames(dist.tab_SW_FB)
colnames(dist.tab_SW_FB)[1] <- "distance"

identical(rownames(metadat_rel), dist.tab_SW_FB$Sample_names)
    #TRUE

dist.tab_SW_FB$Healthy_Diseased_Recovered <- metadat_rel$Healthy_Diseased_Recovered
        #     table(metadat_rel$Disease_Phase)
        #     Disease_discovered            Endemic           Epidemic 
        #                  5                 13                 28 
dist.tab_SW_FB$Date <- metadat_rel$Date
dist.tab_SW_FB$Disease_Phase <- metadat_rel$Disease_Phase
dist.tab_SW_FB$Tag <- metadat_rel$Tag


df <- aggregate(distance ~ Disease_Phase + Healthy_Diseased_Recovered, 
  data=dist.tab_SW_FB, 
  function(x) { 
    c(SD=sd(x), Mean=mean(x)) 
})
df_meanSD_meta <- as.data.frame(df[,1:2])
df_meanSD <- as.data.frame(df$distance)
df2 <- cbind(df_meanSD_meta, df_meanSD)

# I am going to add a date that's in the middle of each phase. This is arbitrary and for plotting purposes. I will add  notation in Illustrator that the mean is across the entire disease phase. 
df2$Date_forPlot <- c("2023-06-01", #Endemic
                      "2021-06-30", #Epidemic
                      "2023-06-01", #Endemic
                      "2021-06-01", #Epidemic
                      "2020-07-16", #Disease_discovered
                      "2021-06-01", #Epidemic
                      "2023-06-01", #Endemic
                      "2021-05-01" #Epidemic
                      )
df2$Date_forPlot <- as.Date(df2$Date_forPlot) #tell R it's a date
```

plotting
```{r}
theme(text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17), 
        legend.position = "bottom", 
        legend.direction ="vertical", 
        plot.margin = unit(c(0, 0.5, 0.5, 0), "cm"))


ggplot() + 
 geom_point(data=prev_fishbay, aes(x = Date_nice, y=Prev_Percent/10), na.rm = TRUE, color="Red", size=2.4) +
  geom_smooth(data=prev_fishbay, aes(x = Date_nice, y=Prev_Percent/10), na.rm = TRUE, method = loess, color="Red", se = FALSE) +
  xlab("") +
  scale_y_continuous(name = "Distance to Centroid (colored circles)",
    sec.axis = sec_axis(~.*10, name="Percent SCTLD Prevelance (red line and red points)")) +
geom_point(data = df2, aes(x=Date_forPlot, y=Mean, color=Healthy_Diseased_Recovered), size=5) +
  geom_errorbar(data = df2, aes(x=Date_forPlot, y=Mean, ymin=Mean-SD, ymax=Mean+SD, color=Healthy_Diseased_Recovered), width=50, size=0.75,) +
  theme_classic() +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered", "Dead"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "gray50")) +
  theme(legend.position = "none", 
        text=element_text(size=17),  
        legend.text=element_text(size=17), 
        panel.grid = element_blank(), 
        axis.text = element_text(size = 17))
```

significance testing for beta diversity in the different disease phases. I want to test sig dif for HDR in epidemic phases, then again for the endemic phase. Testing will need to be nested by Tag (colony ID).

epidemic phase: 
```{r}
#using: dist.tab_SW_FB

dist.tab_epidemic <- subset(dist.tab_SW_FB, Disease_Phase == "Epidemic")
dist.tab_epidemic$Tag <- as.factor(dist.tab_epidemic$Tag)
str(dist.tab_epidemic$Tag)

dist.tab_epidemic$Healthy_Diseased_Recovered <- factor(dist.tab_epidemic$Healthy_Diseased_Recovered, 
      levels = c("Healthy", "Diseased", "Recovered", "Dead"))

#nested ANOVA with Tukey HSD
anova_model <- aov(distance ~ Healthy_Diseased_Recovered / Tag, data = dist.tab_epidemic)
summary(anova_model)
    #                                Df Sum Sq Mean Sq F value   Pr(>F)    
    # Healthy_Diseased_Recovered      3 0.5687 0.18956  10.005 0.000717 ***
    # Healthy_Diseased_Recovered:Tag  9 0.3433 0.03815   2.013 0.110949    
    # Residuals                      15 0.2842 0.01895    
TukeyHSD(anova_model)
#                          diff        lwr          upr     p adj
# Diseased-Healthy    0.3099175  0.1229013  0.496933679 0.0012513
# Recovered-Healthy   0.0147602 -0.1780117  0.207532084 0.9960380
# Dead-Healthy       -0.0101110 -0.3202423  0.300020274 0.9996886
# Recovered-Diseased -0.2951573 -0.4879292 -0.102385399 0.0025307
# Dead-Diseased      -0.3200285 -0.6301598 -0.009897209 0.0420351
# Dead-Recovered     -0.0248712 -0.3385069  0.288764492 0.9956056
```


endemic phase:
```{r}
#endemic
dist.tab_endemic <- subset(dist.tab_SW_FB, Disease_Phase == "Endemic")
dist.tab_endemic$Tag <- as.factor(dist.tab_endemic$Tag)
str(dist.tab_endemic$Tag)

dist.tab_endemic$Healthy_Diseased_Recovered <- factor(dist.tab_endemic$Healthy_Diseased_Recovered, 
      levels = c("Diseased", "Recovered", "Dead"))


anova_model <- aov(distance ~ Healthy_Diseased_Recovered / Tag, data = dist.tab_endemic)
summary(anova_model)
      #                                Df   Sum Sq  Mean Sq F value Pr(>F)
      # Healthy_Diseased_Recovered      2 0.009020 0.004510   1.109  0.399
      # Healthy_Diseased_Recovered:Tag  5 0.006741 0.001348   0.331  0.875
      # Residuals                       5 0.020342 0.004068    
TukeyHSD(anova_model)
      #                            diff        lwr        upr     p adj
      # Recovered-Diseased -0.053896612 -0.1809923 0.07319908 0.4174414
      # Dead-Diseased       0.001220369 -0.2308236 0.23326430 0.9998386
      # Dead-Recovered      0.055116981 -0.1650192 0.27525318 0.7110772
```


## Means: SW from Coral Bay
```{r}
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.seawater.Fishbay", "ps.Seawater.Coralbay", "Prevelance_ForAnalyses", "prev_coralbay", "prev_fishbay")))

# using: ps.Seawater.Coralbay

otu_rel <- as.data.frame(otu_table(ps.Seawater.Coralbay)) # pull out OTU table
otu_rel[1:3,1:3] # make sure sample names are rows
metadat_rel <- as.data.frame(sample_data(ps.Seawater.Coralbay)) # pull out Meta table

# Check if my rows are in the same order
identical(rownames(otu_rel), rownames(metadat_rel)) 
# Should be TRUE

# Make a distance matrix, specify method - bray curtis 
Veg <- vegdist(otu_rel, method="bray") 

# Model to test differences in distance to the centroid within groups in Treatment column
Mod_BetaDisp <- betadisper(Veg, metadat_rel$Healthy_Diseased_Recovered)

Mod_BetaDisp
summary(Mod_BetaDisp)

Mod_aov <- anova(Mod_BetaDisp)
Mod_aov
      # Response: Distances
      #           Df   Sum Sq  Mean Sq F value   Pr(>F)   
      # Groups     2 0.073214 0.036607  7.0557 0.002281 **
      # Residuals 42 0.217910 0.005188   
      #this shows that HDR has an significant effect on beta div. 
      

##################################################

## Prepare for plotting 

dist.tab_SW_CB <- as.data.frame(Mod_BetaDisp$distances)
dist.tab_SW_CB$Sample_names <- rownames(dist.tab_SW_CB)
colnames(dist.tab_SW_CB)[1] <- "distance"

identical(rownames(metadat_rel), dist.tab_SW_CB$Sample_names) 
# Should be TRUE

dist.tab_SW_CB$Healthy_Diseased_Recovered <- metadat_rel$Healthy_Diseased_Recovered
dist.tab_SW_CB$Date <- metadat_rel$Date
dist.tab_SW_CB$Disease_Phase <- metadat_rel$Disease_Phase

######### Code from: 
# http://www.sthda.com/english/wiki/ggplot2-error-bars-quick-start-guide-r-software-and-data-visualization#google_vignette

# Function to calculate the mean and the standard deviation for each group
data_summary <- function(data, varname, groupnames) {
  require(plyr)
  summary_func <- function(x, col) {
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum <- ddply(data, groupnames, .fun=summary_func, varname)
  data_sum <- rename(data_sum, c("mean" = varname))
  return(data_sum)
}

# Summarize the data
df2 <- data_summary(dist.tab_SW_CB, varname="distance", 
                    groupnames=c("Healthy_Diseased_Recovered", "Disease_Phase"))

# Arbitrary date assignment for plotting purposes
df2$Date_forPlot <- c("2023-06-01", # Endemic
                      "2021-06-30", # Epidemic
                      "2021-06-01", # Epidemic
                      "2020-07-16", # Disease discovered
                      "2023-06-01", # Endemic
                      "2021-05-01" # Epidemic
                      )
df2$Date_forPlot <- as.Date(df2$Date_forPlot)

ggplot() + 
 geom_point(data=prev_coralbay, aes(x = Date_nice, y=Prev_Percent/10), na.rm = TRUE, color="Red", size=2.4) +
  geom_smooth(data=prev_coralbay, aes(x = Date_nice, y=Prev_Percent/10), na.rm = TRUE, method = loess, color="Red", se = FALSE) +
  xlab("Date") +
  theme(text=element_text(size=13)) +
  scale_y_continuous(name = "Distance to Centroid (colored shapes)",
    sec.axis = sec_axis(~.*10, name="Percent SCTLD Prevalence (red line and red points)")) +
geom_point(data = df2, aes(x=Date_forPlot, y=distance, color=Healthy_Diseased_Recovered, shape=Disease_Phase), size=5) +
  geom_errorbar(data = df2, aes(x=Date_forPlot, y=distance, ymin=distance-sd, ymax=distance+sd, color=Healthy_Diseased_Recovered), width=50, size=0.75) +
  theme_classic() +
  scale_color_manual(breaks=c("Healthy", "Diseased", "Recovered", "Dead"), 
                       values = c("mediumseagreen", "darkorange2", "darkorchid3", "gray50"))

```



#Stacked Bar plots

##rel abun >= 1% for Seawater from Fish Bay
- code is almost entirely from Chapt GPT
- at first, I tried plotting the ASVs that make up 90% cumulative relative abundance for each sample. But that gave me a total of 1175 unique ASVs, which is more than the number of colors in R, and would not be useful for plotitng.
```{r}
abundance_table <- as.data.frame(otu_table(ps.seawater.Fishbay))
  #46 obs of 7023 variables
abundance_table <- abundance_table[, colSums(abundance_table != 0) > 0]
  #46 obs of 7023 variables

meta_SW.FishBay <- as.data.frame(sample_data(ps.seawater.Fishbay))
data_frame_merge <- merge(abundance_table, meta_SW.FishBay,
                          by = 'row.names', all = TRUE)

# Melt the abundance table to long format for ggplot2
abundance_long <- data_frame_merge %>%
  pivot_longer(cols=starts_with("ASV"), 
               names_to = "ASV_ID", 
               values_to = "Abundance") %>%
  arrange(Row.names, desc(Abundance))
dim(abundance_long) #323058     27

# Filter ASVs with relative abundance >= 0.01 (1%)
abundance_long <- abundance_long %>%
  filter(Abundance >= 0.01)
dim(abundance_long) #858  27 --> lovely!

#how many ASVs are left to plot?
length(unique(abundance_long$ASV_ID)) #114
n <- 114 #plug that number in here to a get that number of unique colors to plot
palette <- distinctColorPalette(n)
  
# Create the stacked bar plot
ggplot(abundance_long, aes(x = Date, y = Abundance, fill = ASV_ID)) +
  geom_bar(stat = "identity", position="stack") +
  facet_grid(~Tag, scale="free") +
 # theme_minimal() +
  scale_fill_manual(values=palette) +
  labs(y = "Relative Abundance", fill = "ASV") +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, hjust = 1),
        legend.position = "none") +
  labs(title = "ASVs with at least 1% relative abundance from Seawater from Fish Bay") +
  xlab(" ")


rm(abundance_table)
rm(meta_SW.FishBay)
rm(data_frame_merge)
rm(abundance_long)
rm(n)
rm(palette)
```

##rel abun >= 1% for Corals from Fish Bay
```{r}
abundance_table <- as.data.frame(otu_table(ps.coral.Fishbay))
  #41 obs of 3871 variables
abundance_table <- abundance_table[, colSums(abundance_table != 0) > 0]
  #41 obs of 3871 variables

meta_Coral.FishBay <- as.data.frame(sample_data(ps.coral.Fishbay))
data_frame_merge <- merge(abundance_table, meta_Coral.FishBay,
                          by = 'row.names', all = TRUE)

# Melt the abundance table to long format for ggplot2
abundance_long <- data_frame_merge %>%
  pivot_longer(cols=starts_with("ASV"), 
               names_to = "ASV_ID", 
               values_to = "Abundance") %>%
  arrange(Row.names, desc(Abundance))
dim(abundance_long) #158711     27

# Filter ASVs with relative abundance >= 0.01 (1%)
abundance_long <- abundance_long %>%
  filter(Abundance >= 0.01)
dim(abundance_long) #687  27 --> lovely!

#how many ASVs are left to plot?
length(unique(abundance_long$ASV_ID)) #544
n <- 544 #plug that number in here to a get that number of unique colors to plot
palette <- distinctColorPalette(n)
  
# Create the stacked bar plot
ggplot(abundance_long, aes(x = Date, y = Abundance, fill = ASV_ID)) +
  geom_bar(stat = "identity", position="stack") +
  facet_grid(~Tag, scale="free") +
 # theme_minimal() +
  scale_fill_manual(values=palette) +
  labs(y = "Relative Abundance", fill = "ASV") +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, hjust = 1),
        legend.position = "none") +
  labs(title = "ASVs with at least 1% relative abundance from Coral from Fish Bay") +
  xlab(" ")


rm(abundance_table)
rm(meta_Coral.FishBay)
rm(data_frame_merge)
rm(abundance_long)
rm(n)
rm(palette)
```

##rel abun >= 1% for Corals from Coral Bay
```{r}
abundance_table <- as.data.frame(otu_table(ps.Coral.Coralbay))
  #35 obs of 5140 variables
abundance_table <- abundance_table[, colSums(abundance_table != 0) > 0]
  #35 obs of 5140 variables

meta_Coral.CoralBay <- as.data.frame(sample_data(ps.Coral.Coralbay))
data_frame_merge <- merge(abundance_table, meta_Coral.CoralBay,
                          by = 'row.names', all = TRUE)

# Melt the abundance table to long format for ggplot2
abundance_long <- data_frame_merge %>%
  pivot_longer(cols=starts_with("ASV"), 
               names_to = "ASV_ID", 
               values_to = "Abundance") %>%
  arrange(Row.names, desc(Abundance))
dim(abundance_long) #179900     27

# Filter ASVs with relative abundance >= 0.01 (1%)
abundance_long <- abundance_long %>%
  filter(Abundance >= 0.01)
dim(abundance_long) #525  27 --> lovely!

#how many ASVs are left to plot?
length(unique(abundance_long$ASV_ID)) #459
n <- 459 #plug that number in here to a get that number of unique colors to plot
palette <- distinctColorPalette(n)
  
# Create the stacked bar plot
ggplot(abundance_long, aes(x = Date, y = Abundance, fill = ASV_ID)) +
  geom_bar(stat = "identity", position="stack") +
  facet_grid(~Tag, scale="free") +
 # theme_minimal() +
  scale_fill_manual(values=palette) +
  labs(y = "Relative Abundance", fill = "ASV") +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, hjust = 1),
        legend.position = "none") +
  labs(title = "ASVs with at least 1% relative abundance from Coral from Coral Bay") +
  xlab(" ")


rm(abundance_table)
rm(meta_Coral.CoralBay)
rm(data_frame_merge)
rm(abundance_long)
rm(n)
rm(palette)
```

##rel abun >= 1% for Seawater from Coral Bay
```{r}
abundance_table <- as.data.frame(otu_table(ps.Seawater.Coralbay))
  #45 obs of 9158 variables
abundance_table <- abundance_table[, colSums(abundance_table != 0) > 0]
  #45 obs of 9158 

meta_Seawater.CoralBay <- as.data.frame(sample_data(ps.Seawater.Coralbay))
data_frame_merge <- merge(abundance_table, meta_Seawater.CoralBay,
                          by = 'row.names', all = TRUE)

# Melt the abundance table to long format for ggplot2
abundance_long <- data_frame_merge %>%
  pivot_longer(cols=starts_with("ASV"), 
               names_to = "ASV_ID", 
               values_to = "Abundance") %>%
  arrange(Row.names, desc(Abundance))
dim(abundance_long) #412110     27

# Filter ASVs with relative abundance >= 0.01 (1%)
abundance_long <- abundance_long %>%
  filter(Abundance >= 0.01)
dim(abundance_long) #859  27 --> lovely!

#how many ASVs are left to plot?
length(unique(abundance_long$ASV_ID)) #73
n <- 73 #plug that number in here to a get that number of unique colors to plot
palette <- distinctColorPalette(n)
  
# Create the stacked bar plot
ggplot(abundance_long, aes(x = Date, y = Abundance, fill = ASV_ID)) +
  geom_bar(stat = "identity", position="stack") +
  facet_grid(~Tag, scale="free") +
 # theme_minimal() +
  scale_fill_manual(values=palette) +
  labs(y = "Relative Abundance", fill = "ASV") +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, hjust = 1),
        legend.position = "none") +
  labs(title = "ASVs with at least 1% relative abundance from Seawater from Coral Bay") +
  xlab(" ")


```




##20 most abundant AVS -- Corals from Fish Bay
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.coral.Fishbay", "ps.Coral.Coralbay", "taxa")))
```{r}
# Subset the phyloseq object by health category
ps_healthy <- subset_samples(ps.coral.Fishbay, Healthy_Diseased_Recovered == "Healthy")
ps_diseased <- subset_samples(ps.coral.Fishbay, Healthy_Diseased_Recovered == "Diseased")
ps_recovered <- subset_samples(ps.coral.Fishbay, Healthy_Diseased_Recovered == "Recovered")

# Function to get top 20 ASVs based on abundance
get_top_ASVs <- function(ps_obj, n = 20) {
  # Aggregate ASV abundance
  ASV_abundance <- taxa_sums(ps_obj)
  # Sort ASVs by abundance and get top n
  top_ASVs <- names(sort(ASV_abundance, decreasing = TRUE)[1:n])
  return(top_ASVs)
}

# Get top 20 ASVs for each health category
top_ASVs_healthy <- get_top_ASVs(ps_healthy)
top_ASVs_diseased <- get_top_ASVs(ps_diseased)
top_ASVs_recovered <- get_top_ASVs(ps_recovered)

# Combine lists and remove duplicates
combined_top_ASVs <- unique(c(top_ASVs_healthy, top_ASVs_diseased, top_ASVs_recovered))
length(combined_top_ASVs) #53

```

```{r}
# Get unique values in the specified category
taxa_topASVs <- as.data.frame(taxa[combined_top_ASVs,])
write.csv(taxa_topASVs, "Top_ASV_FishBay_Corals.csv")
 
# Prune the phyloseq object to keep only the ASVs in combined_top_ASVs
ps_top_ASVs <- prune_taxa(combined_top_ASVs, ps.coral.Fishbay)

# Convert phyloseq object to a data frame using psmelt
df_ps_top_ASVs <- psmelt(ps_top_ASVs)
#see which ASVs appear the most
df_subset <- df_ps_top_ASVs[df_ps_top_ASVs$Abundance > 0, ]
value_counts <- table(df_subset$OTU)

mean_per_sample_FB_C <- df_ps_top_ASVs %>% 
                  group_by(Sample) %>%
                  summarise(mean = mean(Abundance),
                            SD = sd(Abundance),
                            sum = sum(Abundance))
    # min sum =  0.05179833, max sum = 0.94536763

total_mean_FB <- mean_per_sample_FB_C %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.4297116, SD = 0.257606

#set the colors
n <- 53 #plug that number in here to a get that number of unique colors to plot
set.seed(15)
palette <- distinctColorPalette(n)
#pie(rep(1, n), col=palette)


# Reorder OTU levels by decreasing abundance
df_ps_top_ASVs$OTU <- factor(df_ps_top_ASVs$OTU, levels = df_ps_top_ASVs %>%
                          group_by(OTU) %>%
                          summarize(TotalAbundance = sum(Abundance)) %>%
                          arrange(desc(TotalAbundance)) %>%
                          pull(OTU))

plot <- ggplot(df_ps_top_ASVs, aes(x=Date, y=Abundance, fill=OTU)) + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Tag, scale="free") +
  scale_fill_manual(values=palette) +
  theme_bw() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  labs(title = "Top 20 ASVs from Corals from each health category from Fish Bay") +
  xlab(" ")
#%#%#%#%

#get the plot without the legend
plot + theme(legend.position = "none")

#get the legend without the plot
legend_topASVs <- get_legend(plot)
legend <- as.ggplot(legend_topASVs)



```
Barplot_TopASVs_Coral_FishBay
Barplot_Legend_TopASVs_Coral_FishBay





##20 most abundant AVS  -- Seawater from Fish Bay
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.seawater.Fishbay", "ps.Seawater.Coralbay", "taxa")))
```{r}
# Subset the phyloseq object by health category
ps_healthy <- subset_samples(ps.seawater.Fishbay, Healthy_Diseased_Recovered == "Healthy")
ps_diseased <- subset_samples(ps.seawater.Fishbay, Healthy_Diseased_Recovered == "Diseased")
ps_recovered <- subset_samples(ps.seawater.Fishbay, Healthy_Diseased_Recovered == "Recovered")
ps_dead <- subset_samples(ps.seawater.Fishbay, Healthy_Diseased_Recovered == "Dead")

# Function to get top 20 ASVs based on abundance
get_top_ASVs <- function(ps_obj, n = 20) {
  # Aggregate ASV abundance
  ASV_abundance <- taxa_sums(ps_obj)
  # Sort ASVs by abundance and get top n
  top_ASVs <- names(sort(ASV_abundance, decreasing = TRUE)[1:n])
  return(top_ASVs)
}

# Get top 20 ASVs for each health category
top_ASVs_healthy <- get_top_ASVs(ps_healthy)
top_ASVs_diseased <- get_top_ASVs(ps_diseased)
top_ASVs_recovered <- get_top_ASVs(ps_recovered)
top_ASVs_dead <- get_top_ASVs(ps_dead)

# Combine lists and remove duplicates
combined_top_ASVs <- unique(c(top_ASVs_healthy, top_ASVs_diseased, top_ASVs_recovered, top_ASVs_dead))
length(combined_top_ASVs) #35
```

```{r}
# Get unique values in the specified category
taxa_topASVs <- as.data.frame(taxa[combined_top_ASVs,])
write.csv(taxa_topASVs, "Top_ASV_FishBay_Seawater.csv")
 
# Prune the phyloseq object to keep only the ASVs in combined_top_ASVs
ps_top_ASVs <- prune_taxa(combined_top_ASVs, ps.seawater.Fishbay)

# Convert phyloseq object to a data frame using psmelt
df_ps_top_ASVs <- psmelt(ps_top_ASVs)
#see which ASVs appear the most
df_subset <- df_ps_top_ASVs[df_ps_top_ASVs$Abundance > 0, ]
value_counts <- table(df_subset$OTU)
total_by_date <- subset(df_subset, OTU %in% c("ASV45", "ASV84", "ASV79", "ASV67"))
  in1539 <- subset(total_by_date, Date == "2020-12-09" & Tag == "1539")
  sum(in1539$Abundance) #0.620595
  in1587 <- subset(total_by_date, Date == "2021-01-21" & Tag == "1587")
  sum(in1587$Abundance) #0.2718574
  in2749 <- subset(total_by_date, Date == "2021-01-06" & Tag == "2749")
  sum(in2749$Abundance) #0.1762832
  in2749_late <- subset(total_by_date, Date == "2023-08-02" & Tag == "2749")
  sum(in2749_late$Abundance) #0.1762832


  
  
mean_per_sample_FB_SW <- df_ps_top_ASVs %>% 
                  group_by(Sample) %>%
                  summarise(mean = mean(Abundance),
                            SD = sd(Abundance),
                            sum = sum(Abundance))
    # min sum =  0.4503387, max sum = 0.7192226

total_mean_FB <- mean_per_sample_FB_SW %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.6012886, SD = 0.0656139


n <- 35 #plug that number in here to a get that number of unique colors to plot
set.seed(42) 
palette <- distinctColorPalette(n)
#pie(rep(1, n), col=palette)


# Reorder OTU levels by decreasing abundance
df_ps_top_ASVs$OTU <- factor(df_ps_top_ASVs$OTU, levels = df_ps_top_ASVs %>%
                          group_by(OTU) %>%
                          summarize(TotalAbundance = sum(Abundance)) %>%
                          arrange(desc(TotalAbundance)) %>%
                          pull(OTU))


plot <- ggplot(df_ps_top_ASVs, aes(x=Date, y=Abundance, fill=OTU)) + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Tag, scale="free") +
  scale_fill_manual(values=palette) +
  theme_bw() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  labs(title = "Top 20 ASVs from Seawater from each health category from Fish Bay") +
  xlab(" ")
#%#%#%#%

#get the plot without the legend
plot + theme(legend.position = "none")

#get the legend without the plot
legend_topASVs <- get_legend(plot)
legend <- as.ggplot(legend_topASVs)

```
Barplot_TopASVs_Seawater_FishBay
Barplot_Legend_TopASVs_Seawater_FishBay




##20 most abundant AVS -- Corals from Coral Bay
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.Coral.Coralbay", "taxa", "mean_per_sample_FB_C")))
```{r}
# Subset the phyloseq object by health category
ps_before <- subset_samples(ps.Coral.Coralbay, Site_Disease == "Before")
ps_after <- subset_samples(ps.Coral.Coralbay, Site_Disease == "After")

# Function to get top 20 ASVs based on abundance
get_top_ASVs <- function(ps_obj, n = 20) {
  # Aggregate ASV abundance
  ASV_abundance <- taxa_sums(ps_obj)
  # Sort ASVs by abundance and get top n
  top_ASVs <- names(sort(ASV_abundance, decreasing = TRUE)[1:n])
  return(top_ASVs)
}

# Get top 20 ASVs for each health category
top_ASVs_before <- get_top_ASVs(ps_before)
top_ASVs_after <- get_top_ASVs(ps_after)

# Combine lists and remove duplicates
combined_top_ASVs <- unique(c(top_ASVs_before, top_ASVs_after))
length(combined_top_ASVs) #38
```

```{r}
# Get unique values in the specified category
taxa_topASVs <- as.data.frame(taxa[combined_top_ASVs,])
write.csv(taxa_topASVs, "Top_ASV_CoralBay_Corals.csv")
 
# Prune the phyloseq object to keep only the ASVs in combined_top_ASVs
ps_top_ASVs <- prune_taxa(combined_top_ASVs, ps.Coral.Coralbay)

# Convert phyloseq object to a data frame using psmelt
df_ps_top_ASVs <- psmelt(ps_top_ASVs)
#see which ASVs appear the most
df_subset <- df_ps_top_ASVs[df_ps_top_ASVs$Abundance > 0, ]
value_counts <- table(df_subset$OTU)


mean_per_sample_CB_C <- df_ps_top_ASVs %>% 
                  group_by(Sample) %>%
                  summarise(mean = mean(Abundance),
                            SD = sd(Abundance),
                            sum = sum(Abundance))
    # min sum =  0.01202149, max sum = 0.84977306

total_mean_CB <- mean_per_sample_CB_C %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.2843268, SD = 0.265159


n <- 38 #plug that number in here to get that number of unique colors to plot
set.seed(15)
palette <- distinctColorPalette(n)
#pie(rep(1, n), col=palette)

# Reorder OTU levels by decreasing abundance
df_ps_top_ASVs$OTU <- factor(df_ps_top_ASVs$OTU, levels = df_ps_top_ASVs %>%
                          group_by(OTU) %>%
                          summarize(TotalAbundance = sum(Abundance)) %>%
                          arrange(desc(TotalAbundance)) %>%
                          pull(OTU))

plot <- ggplot(df_ps_top_ASVs, aes(x=Date, y=Abundance, fill=OTU)) + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Tag, scale="free") +
  scale_fill_manual(values=palette) +
  theme_bw() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  labs(title = "Top 20 ASVs from Corals from each reef health cat. from Coral Bay") +
  xlab(" ") 
#%#%#%#%

#get the plot without the legend
plot + theme(legend.position = "none")

#get the legend without the plot
legend_topASVs <- get_legend(plot)
legend <- as.ggplot(legend_topASVs)

```
Barplot_TopASVs_Coral_CoralBay
Barplot_Legend_TopASVs_Coral_CoralBay

df_combined <- rbind(mean_per_sample_CB_C, mean_per_sample_CB_C)
total_mean_corals <- df_combined %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
output: 
mean: 0.2843268
SD: 0.2624394
  




##20 most abundant AVS -- Seawater from Coral Bay
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.Seawater.Coralbay", "taxa")))
```{r}
# Subset the phyloseq object by health category
ps_healthy <- subset_samples(ps.Seawater.Coralbay, Healthy_Diseased_Recovered == "Healthy")
ps_diseased <- subset_samples(ps.Seawater.Coralbay, Healthy_Diseased_Recovered == "Diseased")
ps_dead <- subset_samples(ps.Seawater.Coralbay, Healthy_Diseased_Recovered == "Dead")

# Function to get top 20 ASVs based on abundance
get_top_ASVs <- function(ps_obj, n = 20) {
  # Aggregate ASV abundance
  ASV_abundance <- taxa_sums(ps_obj)
  # Sort ASVs by abundance and get top n
  top_ASVs <- names(sort(ASV_abundance, decreasing = TRUE)[1:n])
  return(top_ASVs)
}

# Get top 20 ASVs for each health category
top_ASVs_healthy <- get_top_ASVs(ps_healthy)
top_ASVs_diseased <- get_top_ASVs(ps_diseased)
top_ASVs_dead <- get_top_ASVs(ps_dead)

# Combine lists and remove duplicates
combined_top_ASVs <- unique(c(top_ASVs_healthy, top_ASVs_diseased, top_ASVs_dead))
length(combined_top_ASVs) #31
```

```{r}
# Get unique values in the specified category
taxa_topASVs <- as.data.frame(taxa[combined_top_ASVs,])
write.csv(taxa_topASVs, "Top_ASV_CoralBay_Seawater.csv")
 
# Prune the phyloseq object to keep only the ASVs in combined_top_ASVs
ps_top_ASVs <- prune_taxa(combined_top_ASVs, ps.Seawater.Coralbay)

# Convert phyloseq object to a data frame using psmelt
df_ps_top_ASVs <- psmelt(ps_top_ASVs)

# onlyMay27 <- subset(df_ps_top_ASVs, Date == "2021-05-27")
# onlyMay27_ASVs <- subset(onlyMay27, OTU %in% c("ASV21", "ASV17"))
# mean_onlyMay27_ASVs <- onlyMay27_ASVs %>% 
#                   group_by(OTU) %>%
#                   summarise(mean = mean(Abundance),
#                             SD = sd(Abundance),
#                             sum = sum(Abundance))
#   
only_ASVs <- subset(df_ps_top_ASVs, OTU %in% c("ASV21", "ASV17", "ASV24"))
mean_only_ASVs <- only_ASVs %>%
                  group_by(OTU, Date) %>%
                  summarise(mean = mean(Abundance),
                            SD = sd(Abundance))

#___________
mean_per_sample_CB_SW <- df_ps_top_ASVs %>% 
                  group_by(Sample) %>%
                  summarise(mean = mean(Abundance),
                            SD = sd(Abundance),
                            sum = sum(Abundance))
    # min sum =  0.5080383, max sum = 0.7742816

total_mean_CB <- mean_per_sample_CB_SW %>%
              summarise(mean = mean(sum),
                        SD = sd(sum))
    # mean = 0.6446558, SD = 0.07133171


n <- 31 #plug that number in here to get that number of unique colors to plot
set.seed(42)
palette <- distinctColorPalette(n)
#pie(rep(1, n), col=palette)

# Reorder OTU levels by decreasing abundance
df_ps_top_ASVs$OTU <- factor(df_ps_top_ASVs$OTU, levels = df_ps_top_ASVs %>%
                          group_by(OTU) %>%
                          summarize(TotalAbundance = sum(Abundance)) %>%
                          arrange(desc(TotalAbundance)) %>%
                          pull(OTU))


plot <- ggplot(df_ps_top_ASVs, aes(x=Date, y=Abundance, fill=OTU)) + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Tag, scale="free") +
  scale_fill_manual(values=palette) +
  theme_bw() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  labs(title = "Top 20 ASVs from Seawater from each health category from Coral Bay") +
  xlab(" ")
#%#%#%#%

#get the plot without the legend
plot + theme(legend.position = "none")

#get the legend without the plot
legend_topASVs <- get_legend(plot)
legend <- as.ggplot(legend_topASVs)

```
Barplot_TopASVs_Seawater_CoralBay
Barplot_Legend_TopASVs_Seawater_CoralBay


##Mean abundance of key taxa
```{r}
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.seawater.Fishbay", "ps.Seawater.Coralbay", "taxa")))

#first for Fish Bay
asv_ids <- c("ASV1", "ASV5")
# Extract the abundance data from the phyloseq object
otu_data <- otu_table(ps.seawater.Fishbay)
# Ensure OTU table is in a matrix format (required for some phyloseq objects)
if (!taxa_are_rows(ps.seawater.Fishbay)) {
  otu_data <- t(otu_data)
}
# Subset the OTU table for the selected ASVs
selected_asv_data <- otu_data[asv_ids, , drop = FALSE]
# Calculate mean and standard deviation for each ASV across samples
mean_abundances <- apply(selected_asv_data, 1, mean)
sd_abundances <- apply(selected_asv_data, 1, sd)
# Combine the results into a data frame for easier viewing
result <- data.frame(
  ASV = asv_ids,
  Mean_Abundance = mean_abundances,
  SD_Abundance = sd_abundances
)



#Second for Coral Bay
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.seawater.Fishbay", "ps.Seawater.Coralbay", "taxa")))

asv_ids <- c("ASV1", "ASV3")
# Extract the abundance data from the phyloseq object
otu_data <- otu_table(ps.Seawater.Coralbay)
# Ensure OTU table is in a matrix format (required for some phyloseq objects)
if (!taxa_are_rows(ps.Seawater.Coralbay)) {
  otu_data <- t(otu_data)
}
# Subset the OTU table for the selected ASVs
selected_asv_data <- otu_data[asv_ids, , drop = FALSE]
# Calculate mean and standard deviation for each ASV across samples
mean_abundances <- apply(selected_asv_data, 1, mean)
sd_abundances <- apply(selected_asv_data, 1, sd)
# Combine the results into a data frame for easier viewing
result <- data.frame(
  ASV = asv_ids,
  Mean_Abundance = mean_abundances,
  SD_Abundance = sd_abundances
)

```




#Differentail abundance
This is based on code from Mallory, which she modified from Cynthia's code.

First, I will make a phyloseq object with only healthy and diseased samples, because differential abundance is only done on two factors at a time.

###Healthy or diseased: Seawater from Fish Bay 
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.seawater.Fishbay")))
```{r}
#Make a Seawater from Fish Bay phyloseq object with count data, not relative abundance
#subset the phyloseq object to just data for healthy or diseased corals (no dead)
#subset the phyloseq object to only have epidemic corals
ps.Counts.Seawater.Fishbay <- ps.decontam.filt %>%
  subset_samples(Sample_Type == "Seawater") %>% #subset coral samples
  subset_samples(Site == "Fish_Bay") %>%
  subset_taxa(Family !="Mitochondria" & Order !="Chloroplast") %>% 
  subset_samples(Healthy_or_Diseased != "Dead") %>% #remove dead samples
  subset_samples(Disease_Phase == "Epidemic") %>%
  filter_taxa(function(x) sum(x) > 0, TRUE) #remove ASVS with all zeros


#add a column to the meta data for the number of days since the first sampling date
meta_data <- sample_data(ps.Counts.Seawater.Fishbay)
meta_data$Days_Since_First <- as.numeric(as.Date(meta_data$Date) - as.Date("2020-07-16")) 
sample_data(ps.Counts.Seawater.Fishbay) <- meta_data
rm(meta_data)

#double check that worked
meta_temp <- sample_data(ps.Counts.Seawater.Fishbay) 
rm(meta_temp)
str(sample_data(ps.Counts.Seawater.Fishbay)$Days_Since_First)

#Double check that this is count data
otu_table(ps.Counts.Seawater.Fishbay)[1:3, 1:3]
```

```{r}
# Make factors --> not sure if necessary in this scenario 
sample_data(ps.Counts.Seawater.Fishbay)$Healthy_or_Diseased <-   factor(sample_data(ps.Counts.Seawater.Fishbay)$Healthy_or_Diseased,levels=c("Healthy", "Diseased"))

# Corncob: Differential abundance test  --> controlling for date
set.seed(1)
DA.seawater.epidemic <- differentialTest(formula = ~ Healthy_or_Diseased + Days_Since_First, 
                             phi.formula = ~ Healthy_or_Diseased + Days_Since_First,
                             formula_null = ~ Days_Since_First,
                             phi.formula_null = ~ Healthy_or_Diseased + Days_Since_First,
                             test = "Wald", boot = FALSE,
                             data = ps.Counts.Seawater.Fishbay,
                             fdr_cutoff = 0.1)

DA.seawater.epidemic$significant_taxa
length(DA.seawater.epidemic$significant_taxa) #8! that's a super reasonable number! Woohoo!
#"ASV84"   "ASV158"  "ASV203"  "ASV339"  "ASV2666" "ASV5334" "ASV5356" "ASV6089"
plot(DA.seawater.epidemic)

# Corncob: Differential abundance test 
set.seed(1)
DA.seawater.epidemic <- differentialTest(formula = ~ Healthy_or_Diseased + Tag, 
                             phi.formula = ~ Healthy_or_Diseased + Tag,
                             formula_null = ~ Tag,
                             phi.formula_null = ~ Healthy_or_Diseased + Tag,
                             test = "Wald", boot = FALSE,
                             data = ps.Counts.Seawater.Fishbay,
                             fdr_cutoff = 0.1)

DA.seawater.epidemic$significant_taxa
length(DA.seawater.epidemic$significant_taxa) #141! 
# [1] "ASV1"    "ASV2"    "ASV4"    "ASV5"    "ASV6"    "ASV7"    "ASV8"    "ASV9"    "ASV10"   "ASV12"  
#  [11] "ASV13"   "ASV14"   "ASV17"   "ASV18"   "ASV19"   "ASV23"   "ASV24"   "ASV25"   "ASV26"   "ASV27"  
#  [21] "ASV28"   "ASV31"   "ASV32"   "ASV33"   "ASV34"   "ASV35"   "ASV36"   "ASV39"   "ASV40"   "ASV44"  
#  [31] "ASV46"   "ASV49"   "ASV53"   "ASV57"   "ASV62"   "ASV63"   "ASV64"   "ASV65"   "ASV71"   "ASV72"  
#  [41] "ASV76"   "ASV80"   "ASV82"   "ASV83"   "ASV85"   "ASV88"   "ASV90"   "ASV92"   "ASV93"   "ASV95"  
#  [51] "ASV101"  "ASV108"  "ASV113"  "ASV114"  "ASV115"  "ASV116"  "ASV120"  "ASV121"  "ASV124"  "ASV126" 
#  [61] "ASV129"  "ASV141"  "ASV144"  "ASV149"  "ASV150"  "ASV154"  "ASV157"  "ASV162"  "ASV163"  "ASV164" 
#  [71] "ASV168"  "ASV169"  "ASV172"  "ASV175"  "ASV176"  "ASV178"  "ASV180"  "ASV188"  "ASV190"  "ASV194" 
#  [81] "ASV205"  "ASV207"  "ASV212"  "ASV218"  "ASV225"  "ASV228"  "ASV231"  "ASV236"  "ASV237"  "ASV249" 
#  [91] "ASV252"  "ASV258"  "ASV260"  "ASV261"  "ASV262"  "ASV279"  "ASV283"  "ASV295"  "ASV305"  "ASV315" 
# [101] "ASV322"  "ASV325"  "ASV328"  "ASV330"  "ASV333"  "ASV339"  "ASV352"  "ASV353"  "ASV362"  "ASV372" 
# [111] "ASV374"  "ASV379"  "ASV388"  "ASV413"  "ASV418"  "ASV420"  "ASV427"  "ASV437"  "ASV449"  "ASV453" 
# [121] "ASV460"  "ASV510"  "ASV523"  "ASV531"  "ASV533"  "ASV549"  "ASV564"  "ASV598"  "ASV713"  "ASV739" 
# [131] "ASV747"  "ASV802"  "ASV840"  "ASV902"  "ASV905"  "ASV917"  "ASV1119" "ASV1281" "ASV1509" "ASV1706"
# [141] "ASV3437"
plot(DA.seawater.epidemic) # a bunch positive and negative
```


```{r}
# Make factors --> not sure if necessary in this scenario 
sample_data(ps.Counts.Seawater.Fishbay)$Healthy_or_Diseased <-   factor(sample_data(ps.Counts.Seawater.Fishbay)$Healthy_or_Diseased,levels=c("Healthy", "Diseased"))

# Corncob: Differential abundance test 
set.seed(1)
DA.seawater.epidemic <- differentialTest(formula = ~ Healthy_or_Diseased + Tag + Days_Since_First, 
                             phi.formula = ~ Healthy_or_Diseased + Tag + Days_Since_First,
                             formula_null = ~ Tag + Days_Since_First,
                             phi.formula_null = ~ Healthy_or_Diseased + Tag + Days_Since_First,
                             test = "Wald", boot = FALSE,
                             data = ps.Counts.Seawater.Fishbay,
                             fdr_cutoff = 0.1)
DA.seawater.epidemic$significant_taxa
length(DA.seawater.epidemic$significant_taxa) #88
#  [1] "ASV2"    "ASV4"    "ASV5"    "ASV7"    "ASV8"    "ASV9"    "ASV10"   "ASV12"   "ASV13"   "ASV18"   "ASV19"   "ASV23"  
# [13] "ASV24"   "ASV25"   "ASV27"   "ASV28"   "ASV31"   "ASV32"   "ASV33"   "ASV35"   "ASV36"   "ASV39"   "ASV40"   "ASV44"  
# [25] "ASV46"   "ASV49"   "ASV64"   "ASV65"   "ASV71"   "ASV72"   "ASV73"   "ASV80"   "ASV81"   "ASV82"   "ASV88"   "ASV90"  
# [37] "ASV95"   "ASV100"  "ASV101"  "ASV113"  "ASV114"  "ASV115"  "ASV120"  "ASV121"  "ASV124"  "ASV126"  "ASV149"  "ASV154" 
# [49] "ASV157"  "ASV161"  "ASV168"  "ASV172"  "ASV178"  "ASV180"  "ASV188"  "ASV190"  "ASV194"  "ASV207"  "ASV212"  "ASV218" 
# [61] "ASV236"  "ASV237"  "ASV249"  "ASV258"  "ASV260"  "ASV261"  "ASV262"  "ASV297"  "ASV325"  "ASV328"  "ASV330"  "ASV339" 
# [73] "ASV353"  "ASV362"  "ASV372"  "ASV374"  "ASV379"  "ASV383"  "ASV388"  "ASV413"  "ASV420"  "ASV449"  "ASV510"  "ASV518" 
# [85] "ASV530"  "ASV802"  "ASV902"  "ASV905"  "ASV1119"
plot(DA.seawater.epidemic) # a bunch positive and negative
  #12 positive, 9 have 95% confidence interval that don't cross zero


extractmods2 <- function(model) {
  result <- data.frame("Estimate" = model$coefficients[2, 1], 
                        "Std.Error" = model$coefficients[2, 2], 
                        "p" = model$coefficients[2, 4])
  return(result)
}

# save the output and add asv, p.adj values to it and make it into a data frame
da.models <- lapply(DA.seawater.epidemic$significant_models, extractmods2)
names(da.models) <- DA.seawater.epidemic$significant_taxa
sig_taxa <- map2_dfr(names(da.models), da.models, ~data.frame(
                      ASV = .x,
                      Estimate = .y$Estimate,
                      Std.Error = .y$Std.Error,
                      p = .y$p
                    ))
#pull out the taxa with positive differential abundance
all_positive_sig_df <- sig_taxa %>% filter(Estimate > 0)
length(all_positive_sig_df$ASV) #12
#I am going to just manually remove the taxa whose 95% confident intervals overlap with zero (73, 388, 449)
positive_sig_df <- all_positive_sig_df %>% filter(!ASV %in% c("ASV73", "ASV388", "ASV449"))
length(positive_sig_df$ASV) #9 

```




#ASVs over time
##Seawater from Fish Bay
```{r}
#clean up environment
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.seawater.Fishbay", "ps.Seawater.Coralbay", "positive_sig_df", "asv_abundance")))

asv_abundance <- as.data.frame(as.matrix(otu_table(ps.seawater.Fishbay)))
 #46 obs of 7023 variables
meta_forBubble <- sample_data(ps.seawater.Fishbay)

ASVofInterst <- positive_sig_df[,1]
ASVofInterst <- c("ASV2", ASVofInterst)
asv_abun_ofInterest <- asv_abundance[, ASVofInterst]
asv_abun_ofInterest$Sample_names <- row.names(asv_abun_ofInterest)

combo_data <- cbind(asv_abun_ofInterest, meta_forBubble)

colNeeded <- c("ASV2", "ASV65", "ASV72", "ASV120", "ASV194", "ASV260", "ASV297", "ASV379", "ASV383", "ASV530", "Sample_names", "Site", "Tag", "Date", "Sample_Type", "Healthy_Diseased_Recovered")
combo_data <- combo_data[,colNeeded]
```

```{r}
combo_data_forplotting <- combo_data %>%
  pivot_longer(cols = starts_with("ASV"), 
               names_to = "ASV", 
               values_to = "RelativeAbundance")
combo_data_forplotting$Date_nice <- as.Date(combo_data_forplotting$Date) #make ggplot know it's a date
str(combo_data_forplotting$Date)

ggplot(combo_data_forplotting, aes(x = Date, y = RelativeAbundance, color = ASV, group = ASV)) +
  geom_point() +                       # Adds points for each data point
  geom_line() +                        # Connects points within each ASV group
  facet_wrap(~Tag, scales = "fixed", ncol = 3) +  # Adds x-axis labels to top and bottom rows
  scale_y_continuous(name = "Relative abundance") +
  xlab("") +
  theme_bw() +
  theme(
    text = element_text(size = 16),
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
    scale_color_manual(breaks=c("ASV2", "ASV65", "ASV72", "ASV120", "ASV194", "ASV260", "ASV297", "ASV379", "ASV383", "ASV530"), values = c( "#996FD2", "darkgoldenrod3", "darkgreen", "#DB7BB2", "chartreuse3", "deeppink3", "darkorange3", "cornsilk4", "blue3", "cyan3")) 

```

##Seawater from Coral Bay
```{r}
#clean up environment
rm(list = setdiff(ls(), c("ps.decontam.filt", "ps.seawater.Fishbay", "ps.Seawater.Coralbay", "positive_sig_df", "asv_abundance")))

asv_abundance <- as.data.frame(as.matrix(otu_table(ps.Seawater.Coralbay)))
 #45 obs of 9158 variables
meta_forBubble <- sample_data(ps.Seawater.Coralbay)

ASVofInterst <- c("ASV2", "ASV65", "ASV72", "ASV120", "ASV194", "ASV260", "ASV297", "ASV379", "ASV383", "ASV530")
asv_abun_ofInterest <- asv_abundance[, ASVofInterst]
asv_abun_ofInterest$Sample_names <- row.names(asv_abun_ofInterest)

combo_data <- cbind(asv_abun_ofInterest, meta_forBubble)

colNeeded <- c("ASV2", "ASV65", "ASV72", "ASV120", "ASV194", "ASV260", "ASV297", "ASV379", "ASV383", "ASV530", "Sample_names", "Site", "Tag", "Date", "Sample_Type", "Healthy_Diseased_Recovered", "Site_Disease")
combo_data <- combo_data[,colNeeded]
```


```{r}
combo_data_forplotting <- combo_data %>%
  pivot_longer(cols = starts_with("ASV"), 
               names_to = "ASV", 
               values_to = "RelativeAbundance")
combo_data_forplotting$Date_nice <- as.Date(combo_data_forplotting$Date) #make ggplot know it's a date
str(combo_data_forplotting$Date)

ggplot(combo_data_forplotting, aes(x = Date, y = RelativeAbundance, color = ASV, group = ASV)) +
  geom_point() +                       # Adds points for each data point
  geom_line() +                        # Connects points within each ASV group
  facet_wrap(~Tag, scales = "fixed", ncol = 3) +  # Adds x-axis labels to top and bottom rows
  scale_y_continuous(name = "Relative abundance") +
  xlab("") +
  theme_bw() +
  theme(
    text = element_text(size = 16),
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
    scale_color_manual(breaks=c("ASV2", "ASV65", "ASV72", "ASV120", "ASV194", "ASV260", "ASV297", "ASV379", "ASV383", "ASV530"), values = c( "#996FD2", "darkgoldenrod3", "darkgreen", "#DB7BB2", "chartreuse3", "deeppink3", "darkorange3", "cornsilk4", "blue3", "cyan3")) 

```








